00.10-E-Peve-sRNAseq-reads-FastQC-MultiQC.Rmd
================
Kathleen Durkin
2024-04-30

- <a href="#1-create-a-bash-variables-file"
  id="toc-1-create-a-bash-variables-file">1 Create a Bash variables
  file</a>
- <a href="#2-download-pevermanni-srna-seq-fastqs"
  id="toc-2-download-pevermanni-srna-seq-fastqs">2 Download
  <em>P.evermanni</em> sRNA-seq FastQs</a>
  - <a href="#21-download-raw-srna-seq-reads"
    id="toc-21-download-raw-srna-seq-reads">2.1 Download raw sRNA-seq
    reads</a>
  - <a href="#22-check-number-of-files"
    id="toc-22-check-number-of-files">2.2 Check number of files</a>
  - <a href="#23-verify-raw-read-checksums"
    id="toc-23-verify-raw-read-checksums">2.3 Verify raw read checksums</a>
- <a href="#3-fastqcmultiqc-on-raw-reads"
  id="toc-3-fastqcmultiqc-on-raw-reads">3 FastQC/MultiQC on raw reads</a>

This Rmd file will download raw RNA-seq FastQs for *P.evermanni* and
evaluate them using [FastQC](https://github.com/s-andrews/FastQC) and
[MultiQC](https://multiqc.info/)(Ewels et al. 2016).

# 1 Create a Bash variables file

This allows usage of Bash variables across R Markdown chunks.

``` bash
{
echo "#### Assign Variables ####"
echo ""

echo "# Data directories"
echo 'export timeseries_dir=/home/shared/8TB_HDD_02/shedurkin/timeseries_molecular'
echo 'export output_dir_top=${timeseries_dir}/E-Peve/output/00.10-E-Peve-sRNAseq-reads-FastQC-MultiQC'
echo 'export raw_fastqc_dir=${output_dir_top}/raw-fastqc-sRNA'
echo 'export raw_reads_dir=${timeseries_dir}/E-Peve/data/raw-fastqs-sRNA'
echo 'export raw_reads_url="https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1069297013/"'
echo ""

echo "# Paths to programs"
echo 'export fastqc=/home/shared/FastQC-0.12.1/fastqc'
echo 'export multiqc=/home/sam/programs/mambaforge/bin/multiqc'

echo ""

echo "# Set FastQ filename patterns"
echo "export fastq_pattern='*.fastq.gz'"
echo "export R1_fastq_pattern='*_R1_*.fastq.gz'"
echo "export R2_fastq_pattern='*_R2_*.fastq.gz'"
echo ""

echo "# Set number of CPUs to use"
echo 'export threads=40'
echo ""


echo "## Inititalize arrays"
echo 'export fastq_array_R1=()'
echo 'export fastq_array_R2=()'
echo 'export raw_fastqs_array=()'
echo 'export R1_names_array=()'
echo 'export R2_names_array=()'
echo ""

echo "# Programs associative array"
echo "declare -A programs_array"
echo "programs_array=("
echo '[fastqc]="${fastqc}" \'
echo '[multiqc]="${multiqc}" \'
echo ")"
echo ""

echo "# Print formatting"
echo 'export line="--------------------------------------------------------"'
echo ""
} > .bashvars

cat .bashvars
```

    #### Assign Variables ####

    # Data directories
    export timeseries_dir=/home/shared/8TB_HDD_02/shedurkin/timeseries_molecular
    export output_dir_top=${timeseries_dir}/E-Peve/output/00.10-E-Peve-sRNAseq-reads-FastQC-MultiQC
    export raw_fastqc_dir=${output_dir_top}/raw-fastqc-sRNA
    export raw_reads_dir=${timeseries_dir}/E-Peve/data/raw-fastqs-sRNA
    export raw_reads_url="https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1069297013/"

    # Paths to programs
    export fastqc=/home/shared/FastQC-0.12.1/fastqc
    export multiqc=/home/sam/programs/mambaforge/bin/multiqc

    # Set FastQ filename patterns
    export fastq_pattern='*.fastq.gz'
    export R1_fastq_pattern='*_R1_*.fastq.gz'
    export R2_fastq_pattern='*_R2_*.fastq.gz'

    # Set number of CPUs to use
    export threads=40

    ## Inititalize arrays
    export fastq_array_R1=()
    export fastq_array_R2=()
    export raw_fastqs_array=()
    export R1_names_array=()
    export R2_names_array=()

    # Programs associative array
    declare -A programs_array
    programs_array=(
    [fastqc]="${fastqc}" \
    [multiqc]="${multiqc}" \
    )

    # Print formatting
    export line="--------------------------------------------------------"

# 2 Download *P.evermanni* sRNA-seq FastQs

## 2.1 Download raw sRNA-seq reads

Reads are downloaded from
<https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1069297013/>

Since sequencing included multiple species, the code will also parse
only those that are *P.evermanni*.

The `--cut-dirs 3` command cuts the preceding directory structure
(i.e. `nightingales/E5-coral-time-series/30-1047560508/`) so that we
just end up with the reads.

``` bash
# Load bash variables into memory
source .bashvars

# Make output directory if it doesn't exist
mkdir --parents ${raw_reads_dir}

# Create list of only P.evermanni sample names
sample_list=$(awk -F "," '$6 ~ /^POR/ {print $5}' ${timeseries_dir}/M-multi-species/data/rna_metadata.csv)

echo ""
echo "${line}"
echo ""
echo "Sample list:"
echo ""
echo "${sample_list}"
echo ""
echo "${line}"
echo ""


# Use printf to format each item for use in wget
formatted_list=$(printf "%s_*," ${sample_list})

# Remove the trailing comma
formatted_list=${formatted_list%,}

# Output the final wget command
echo ""
echo "${line}"
echo ""
echo "Formatted wget accept list:"
echo ""
echo "wget --accept=\"$formatted_list\""
echo ""
echo "${line}"
echo ""

# Run wget to retrieve FastQs and MD5 files
# Note: the --no-clobber command will skip re-downloading any files that are already present in the output directory
wget \
--directory-prefix ${raw_reads_dir} \
--recursive \
--no-check-certificate \
--continue \
--cut-dirs 3 \
--no-host-directories \
--no-parent \
--quiet \
--no-clobber \
--accept=${formatted_list} ${raw_reads_url}

ls -lh "${raw_reads_dir}"
```

## 2.2 Check number of files

We have 38 P.evermanni samples (10 colonies, 4 timepoints each, with the
exception of POR-236 for which we only have samples from TP1 and TP2),
so there should be 76 total sequence files (forward and reverse read for
each sample).

``` bash
# Load bash variables into memory
source .bashvars

echo "Number of fastq files:"
ls ${raw_reads_dir}/${fastq_pattern} | wc -l

echo "Number of md5 files:"
ls ${raw_reads_dir}/*.md5 | wc -l
```

We only have 74 of each instead of the expected 76, indicating one
sample is missing.

After crossreferencing with the metadata df, it looks like we’re missing
data for sample 1E2 (POR-69, TP3). Not clear why we’re missing those
files, made a [github
issue](https://github.com/RobertsLab/resources/issues/2186). For now
I’ll continue with file QC, trimming, etc., and may come back later to
deal with 1E2 if we find the data.

## 2.3 Verify raw read checksums

``` bash
# Load bash variables into memory
source .bashvars

cd "${raw_reads_dir}"

# Checksums file contains other files, so this just looks for the sRNAseq files.
for file in *.md5
do
  md5sum --check "${file}"
done
```

# 3 FastQC/MultiQC on raw reads

``` bash
# Load bash variables into memory
source .bashvars

# Make output directory if it doesn't exist
mkdir --parents "${raw_fastqc_dir}"

############ RUN FASTQC ############


# Create array of trimmed FastQs
raw_fastqs_array=(${raw_reads_dir}/${fastq_pattern})

# Pass array contents to new variable as space-delimited list
raw_fastqc_list=$(echo "${raw_fastqs_array[*]}")

echo "Beginning FastQC on raw reads..."
echo ""

# Run FastQC
### NOTE: Do NOT quote raw_fastqc_list
${programs_array[fastqc]} \
--threads ${threads} \
--outdir ${raw_fastqc_dir} \
--quiet \
${raw_fastqc_list}

echo "FastQC on raw reads complete!"
echo ""

############ END FASTQC ############

############ RUN MULTIQC ############
echo "Beginning MultiQC on raw FastQC..."
echo ""

${programs_array[multiqc]} ${raw_fastqc_dir} -o ${raw_fastqc_dir}

echo ""
echo "MultiQC on raw FastQs complete."
echo ""

############ END MULTIQC ############

echo "Removing FastQC zip files."
echo ""
rm ${raw_fastqc_dir}/*.zip
echo "FastQC zip files removed."
echo ""
```

``` bash
# Load bash variables into memory
source .bashvars

# View directory contents
ls -lh ${raw_fastqc_dir}
```

    total 52M
    -rw-r--r-- 1 shedurkin labmembers 708K Apr 30 14:18 1A11_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 672K Apr 30 14:17 1A11_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 696K Apr 30 14:21 1A7_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 666K Apr 30 14:21 1A7_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 744K Apr 30 14:16 1B3_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 716K Apr 30 14:16 1B3_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 663K Apr 30 14:26 1B4_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 643K Apr 30 14:26 1B4_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 672K Apr 30 14:17 1B6_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 655K Apr 30 14:17 1B6_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 739K Apr 30 14:16 1B7_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 708K Apr 30 14:16 1B7_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 751K Apr 30 14:16 1C11_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 719K Apr 30 14:16 1C11_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 685K Apr 30 14:17 1C12_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 654K Apr 30 14:17 1C12_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 680K Apr 30 14:31 1C5_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 657K Apr 30 14:30 1C5_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 661K Apr 30 14:16 1C6_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 642K Apr 30 14:16 1C6_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 728K Apr 30 14:16 1D12_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 699K Apr 30 14:16 1D12_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 660K Apr 30 14:25 1D1_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 643K Apr 30 14:24 1D1_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 695K Apr 30 14:31 1D2_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 656K Apr 30 14:30 1D2_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 662K Apr 30 14:26 1D5_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 639K Apr 30 14:25 1D5_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 691K Apr 30 14:21 1D7_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 659K Apr 30 14:21 1D7_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 728K Apr 30 14:16 1E7_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 697K Apr 30 14:16 1E7_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 691K Apr 30 14:19 1F10_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 660K Apr 30 14:19 1F10_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 697K Apr 30 14:31 1F12_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 658K Apr 30 14:30 1F12_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 693K Apr 30 14:16 1F1_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 683K Apr 30 14:16 1F1_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 681K Apr 30 14:32 1F3_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 660K Apr 30 14:31 1F3_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 673K Apr 30 14:25 1F6_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 649K Apr 30 14:25 1F6_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 688K Apr 30 14:18 1F7_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 667K Apr 30 14:18 1F7_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 723K Apr 30 14:31 1G10_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 692K Apr 30 14:29 1G10_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 679K Apr 30 14:31 1G1_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 656K Apr 30 14:31 1G1_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 693K Apr 30 14:17 1G3_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 670K Apr 30 14:17 1G3_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 678K Apr 30 14:18 1H1_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 661K Apr 30 14:18 1H1_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 687K Apr 30 14:17 1H2_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 677K Apr 30 14:17 1H2_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 686K Apr 30 14:18 1H3_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 668K Apr 30 14:18 1H3_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 718K Apr 30 14:18 2A3_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 696K Apr 30 14:18 2A3_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 682K Apr 30 14:26 2B1_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 656K Apr 30 14:25 2B1_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 696K Apr 30 14:25 2D1_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 673K Apr 30 14:24 2D1_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 733K Apr 30 14:19 2D3_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 704K Apr 30 14:19 2D3_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 691K Apr 30 14:26 2E1_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 670K Apr 30 14:25 2E1_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 703K Apr 30 14:18 2E3_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 690K Apr 30 14:18 2E3_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 685K Apr 30 14:18 2F2_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 674K Apr 30 14:18 2F2_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 707K Apr 30 14:18 2H1_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 693K Apr 30 14:18 2H1_R2_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 687K Apr 30 14:27 2H2_R1_001_fastqc.html
    -rw-r--r-- 1 shedurkin labmembers 660K Apr 30 14:26 2H2_R2_001_fastqc.html
    drwxr-xr-x 2 shedurkin labmembers 4.0K Apr 30 14:32 multiqc_data
    -rw-r--r-- 1 shedurkin labmembers 2.4M Apr 30 14:32 multiqc_report.html

<div id="refs" class="references csl-bib-body hanging-indent">

<div id="ref-ewels2016" class="csl-entry">

Ewels, Philip, Måns Magnusson, Sverker Lundin, and Max Käller. 2016.
“MultiQC: Summarize Analysis Results for Multiple Tools and Samples in a
Single Report.” *Bioinformatics* 32 (19): 3047–48.
<https://doi.org/10.1093/bioinformatics/btw354>.

</div>

</div>
