---
title: "13.00-multiomics-barnacle"
author: "Sam White"
date: "2025-10-03"
output: 
  github_document:
    toc: true
    number_sections: true
  bookdown::html_document2:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
bibliography: references.bib
citeproc: true
---

# BACKGROUND

# Setup

## Libraries

```{r setup, include=FALSE}
library(knitr)
library(tidyverse)
library(reticulate)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  comment = ""         # Prevents appending '##' to beginning of lines in code output
)
```

## Set R variables

```{r R-variables, eval=TRUE}
# OUTPUT DIRECTORY
output_dir <- "../output/13.00-multiomics-barnacle"

#INPUT FILE(S)
ortholog_groups_file <- "../output/12-ortho-annot/ortholog_groups_annotated.csv"

# Transcript count matrices for each species
apul_transcript_matrix_file <- "../../D-Apul/output/02.20-D-Apul-RNAseq-alignment-HiSat2/apul-transcript_count_matrix.csv"
peve_transcript_matrix_file <- "../../E-Peve/output/02.20-E-Peve-RNAseq-alignment-HiSat2/peve-transcript_count_matrix.csv"
ptua_transcript_matrix_file <- "../../F-Ptua/output/02.20-F-Ptua-RNAseq-alignment-HiSat2/ptua-transcript_count_matrix.csv"

# CONDA
conda_env_name <- c("/home/sam/programs/mambaforge/envs/barnacle_py311_env")
conda_path <- c("/opt/anaconda/anaconda3/bin/conda")
```

## Load [barnacle](https://github.com/blasks/barnacle) conda environment

If this is successful, the first line of output should show that the Python being used is the one in your [barnacle](https://github.com/blasks/barnacle) conda environment path.

E.g.

`python:         /home/sam/programs/mambaforge/envs/barnacle_py311_env/bin/python`

```{r load-barnacle-conda-env, eval=TRUE}
use_condaenv(condaenv = conda_env_name, conda = conda_path)
py_config()
```

# DATA PREP

## Load ortholog groups data

```{r load-ortholog-data, eval=TRUE}
# Read in the ortholog groups data
ortholog_groups <- read.csv(ortholog_groups_file)



# Display basic info about the data
cat("Dimensions of ortholog groups data:", dim(ortholog_groups), "\n\n")
cat("Column names:", colnames(ortholog_groups), "\n\n")
head(ortholog_groups)
str(ortholog_groups)
```

## Read in transcript expression data

```{r extract-gene-expression, eval=TRUE}
# Read in transcript count matrices for each species
cat("Loading transcript count matrices...\n")


apul_transcripts <- read.csv(apul_transcript_matrix_file, row.names = 1)
cat("Apul transcript matrix dimensions:", dim(apul_transcripts), "\n")


peve_transcripts <- read.csv(peve_transcript_matrix_file, row.names = 1)
cat("Peve transcript matrix dimensions:", dim(peve_transcripts), "\n")


ptua_transcripts <- read.csv(ptua_transcript_matrix_file, row.names = 1)
cat("Ptua transcript matrix dimensions:", dim(ptua_transcripts), "\n")

# Investigate differences in column dimensions
cat("\n=== COLUMN DIMENSION ANALYSIS ===\n")
cat("Expected columns: 40\n")
cat("Actual columns:\n")
cat("  Apul:", ncol(apul_transcripts), "\n")
cat("  Peve:", ncol(peve_transcripts), "\n")
cat("  Ptua:", ncol(ptua_transcripts), "\n")

# Show column names for each species
cat("\nApul column names:\n")
print(colnames(apul_transcripts))
cat("\nPeve column names:\n")
print(colnames(peve_transcripts))
cat("\nPtua column names:\n")
print(colnames(ptua_transcripts))

# Compare column names across species
all_cols <- list(
  Apul = colnames(apul_transcripts),
  Peve = colnames(peve_transcripts),
  Ptua = colnames(ptua_transcripts)
)
cat("\n\n")

cat("Structure of apul_transcripts data frame:\n\n")
str(apul_transcripts)
cat("\n\n")

cat("Structure of peve_transcripts data frame:\n\n")
str(peve_transcripts)
cat("\n\n")

cat("Structure of apul_transcripts data frame:\n\n")
str(ptua_transcripts)
```


# REFERENCES