00.10-D-Apul-sRNAseq-reads-FastQC-MultiQC.Rmd
================
Kathleen Durkin
2024-12-18

- [1 Create a Bash variables file](#1-create-a-bash-variables-file)
- [2 Download *A.pulchra* sRNA-seq
  FastQs](#2-download-apulchra-srna-seq-fastqs)
  - [2.1 Download raw sRNA-seq reads](#21-download-raw-srna-seq-reads)
  - [2.2 Check number of files](#22-check-number-of-files)
  - [2.3 Verify raw read checksums](#23-verify-raw-read-checksums)
- [3 FastQC/MultiQC on raw reads](#3-fastqcmultiqc-on-raw-reads)

Note: modified from Sam White’s code in
`00.00-D-Apul-RNAseq-reads-FastQC-MultiQC`

This Rmd file will download raw RNA-seq FastQs for *A.pulchra* and
evaluate them using [FastQC](https://github.com/s-andrews/FastQC) and
[MultiQC](https://multiqc.info/)(Ewels et al. 2016).

# 1 Create a Bash variables file

This allows usage of Bash variables across R Markdown chunks.

``` bash
{
echo "#### Assign Variables ####"
echo ""

echo "# Data directories"
echo 'export timeseries_dir=/home/shared/8TB_HDD_02/shedurkin/timeseries_molecular'
echo 'export output_dir_top=${timeseries_dir}/D-Apul/output/00.10-D-Apul-sRNAseq-reads-FastQC-MultiQC'
echo 'export raw_fastqc_dir=${output_dir_top}/raw-fastqc-sRNA'
echo 'export raw_reads_dir=${timeseries_dir}/D-Apul/data/raw-fastqs-sRNA'
echo 'export raw_reads_url="https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1069297013/"'
echo ""

echo "# Paths to programs"
echo 'export fastqc=/home/shared/FastQC-0.12.1/fastqc'
echo 'export multiqc=/home/sam/programs/mambaforge/bin/multiqc'
echo ""

echo "# Set FastQ filename patterns"
echo "export fastq_pattern='*.fastq.gz'"
echo "export R1_fastq_pattern='*_R1_*.fastq.gz'"
echo "export R2_fastq_pattern='*_R2_*.fastq.gz'"
echo ""

echo "# Set number of CPUs to use"
echo 'export threads=40'
echo ""


echo "## Inititalize arrays"
echo 'export fastq_array_R1=()'
echo 'export fastq_array_R2=()'
echo 'export raw_fastqs_array=()'
echo 'export R1_names_array=()'
echo 'export R2_names_array=()'
echo ""

echo "# Programs associative array"
echo "declare -A programs_array"
echo "programs_array=("
echo '[fastqc]="${fastqc}" \'
echo '[multiqc]="${multiqc}" \'
echo ")"
echo ""

echo "# Print formatting"
echo 'export line="--------------------------------------------------------"'
echo ""
} > .bashvars

cat .bashvars
```

    #### Assign Variables ####

    # Data directories
    export timeseries_dir=/home/shared/8TB_HDD_02/shedurkin/timeseries_molecular
    export output_dir_top=${timeseries_dir}/D-Apul/output/00.10-D-Apul-sRNAseq-reads-FastQC-MultiQC
    export raw_fastqc_dir=${output_dir_top}/raw-fastqc-sRNA
    export raw_reads_dir=${timeseries_dir}/D-Apul/data/raw-fastqs-sRNA
    export raw_reads_url="https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1069297013/"

    # Paths to programs
    export fastqc=/home/shared/FastQC-0.12.1/fastqc
    export multiqc=/home/sam/programs/mambaforge/bin/multiqc

    # Set FastQ filename patterns
    export fastq_pattern='*.fastq.gz'
    export R1_fastq_pattern='*_R1_*.fastq.gz'
    export R2_fastq_pattern='*_R2_*.fastq.gz'

    # Set number of CPUs to use
    export threads=40

    ## Inititalize arrays
    export fastq_array_R1=()
    export fastq_array_R2=()
    export raw_fastqs_array=()
    export R1_names_array=()
    export R2_names_array=()

    # Programs associative array
    declare -A programs_array
    programs_array=(
    [fastqc]="${fastqc}" \
    [multiqc]="${multiqc}" \
    )

    # Print formatting
    export line="--------------------------------------------------------"

# 2 Download *A.pulchra* sRNA-seq FastQs

## 2.1 Download raw sRNA-seq reads

Reads are downloaded from
<https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1069297013/>

Since sequencing included multiple species, the code will also parse
only those that are *A.pulchra*.

The `--cut-dirs 3` command cuts the preceding directory structure
(i.e. `nightingales/E5-coral-time-series/30-1047560508/`) so that we
just end up with the reads.

``` bash
# Load bash variables into memory
source .bashvars

# Make output directory if it doesn't exist
mkdir --parents ${raw_reads_dir}

# Create list of only A.pulchra sample names
sample_list=$(awk -F "," '$6 ~ /^ACR/ {print $5}' ${timeseries_dir}/M-multi-species/data/rna_metadata.csv)

echo ""
echo "${line}"
echo ""
echo "Sample list:"
echo ""
echo "${sample_list}"
echo ""
echo "${line}"
echo ""


# Use printf to format each item for use in wget
formatted_list=$(printf "%s_*," ${sample_list})

# Remove the trailing comma
formatted_list=${formatted_list%,}

# Output the final wget command
echo ""
echo "${line}"
echo ""
echo "Formatted wget accept list:"
echo ""
echo "wget --accept=\"$formatted_list\""
echo ""
echo "${line}"
echo ""

# Run wget to retrieve FastQs and MD5 files
# Note: the --no-clobber command will skip re-downloading any files that are already present in the output directory
wget \
--directory-prefix ${raw_reads_dir} \
--recursive \
--no-check-certificate \
--continue \
--cut-dirs 3 \
--no-host-directories \
--no-parent \
--quiet \
--no-clobber \
--accept=${formatted_list} ${raw_reads_url}

ls -lh "${raw_reads_dir}"
```

## 2.2 Check number of files

We have 40 A.pulchra samples (10 colonies, 4 timepoints each), so there
should be 80 total sequence files (forward and reverse read for each
sample).

``` bash
# Load bash variables into memory
source .bashvars

echo "Number of fastq files:"
ls ${raw_reads_dir}/${fastq_pattern} | wc -l

echo "Number of md5 files:"
ls ${raw_reads_dir}/*.md5 | wc -l
```

We’re good!

## 2.3 Verify raw read checksums

``` bash
# Load bash variables into memory
source .bashvars

cd "${raw_reads_dir}"

# Checksums file contains other files, so this just looks for the sRNAseq files.
for file in *.md5
do
  md5sum --check "${file}"
done
```

# 3 FastQC/MultiQC on raw reads

``` bash
# Load bash variables into memory
source .bashvars

# Make output directory if it doesn't exist
mkdir --parents "${raw_fastqc_dir}"

############ RUN FASTQC ############


# Create array of trimmed FastQs
raw_fastqs_array=(${raw_reads_dir}/${fastq_pattern})

# Pass array contents to new variable as space-delimited list
raw_fastqc_list=$(echo "${raw_fastqs_array[*]}")

echo "Beginning FastQC on raw reads..."
echo ""

# Run FastQC
### NOTE: Do NOT quote raw_fastqc_list
${programs_array[fastqc]} \
--threads ${threads} \
--outdir ${raw_fastqc_dir} \
--quiet \
${raw_fastqc_list}

echo "FastQC on raw reads complete!"
echo ""

############ END FASTQC ############

############ RUN MULTIQC ############
echo "Beginning MultiQC on raw FastQC..."
echo ""

${programs_array[multiqc]} ${raw_fastqc_dir} -o ${raw_fastqc_dir}

echo ""
echo "MultiQC on raw FastQs complete."
echo ""

############ END MULTIQC ############

echo "Removing FastQC zip files."
echo ""
rm ${raw_fastqc_dir}/*.zip
echo "FastQC zip files removed."
echo ""

# View directory contents
ls -lh ${raw_fastqc_dir}
```

<div id="refs" class="references csl-bib-body hanging-indent">

<div id="ref-ewels2016" class="csl-entry">

Ewels, Philip, Måns Magnusson, Sverker Lundin, and Max Käller. 2016.
“MultiQC: Summarize Analysis Results for Multiple Tools and Samples in a
Single Report.” *Bioinformatics* 32 (19): 3047–48.
<https://doi.org/10.1093/bioinformatics/btw354>.

</div>

</div>
