---
title: "00.30-D-Apul-transcriptome-assembly-Trinity"
author: "Sam White"
date: "2025-11-21"
output: 
  bookdown::html_document2:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
  github_document:
    toc: true
    number_sections: true
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
bibliography: references.bib
---
# BACKGROUND


# SETUP


## Libraries and markdown settings
```{r setup, include=TRUE}
library(knitr)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  comment = ""         # Prevents appending '##' to beginning of lines in code output
)
```


## Set variables
```{r set-variables, eval=TRUE}
# DIRECTORIES
top_output_dir <- file.path("..", "output")

output_dir <- file.path(top_output_dir, "00.30-D-Apul-transcriptome-assembly-Trinity")
trimmed_reads_dir <- file.path(output_dir, "output", "01.00-D-Apul-RNAseq-trimming-fastp-FastQC-MultiQC")


# FILES
bam_alignment <- file.path(top_output_dir, "02.20-D-Apul-RNAseq-alignment-HiSat2", "sorted-bams-merged.bam")
genome_fasta <- file.path("..", "data", "Apulchra-genome.fa")
denovo_assembly_name <- "apul-denovo-Trinity"
genome_guided_assembly_name <- "apul-GG-Trinity"

#SETTINGS
## THREADS
threads <- "150"

## MAX RAM
max_ram <- "100G"


# FORMATTING
line <- "-----------------------------------------------"

# Export these as environment variables for bash chunks.
Sys.setenv(
  bam_alignment = bam_alignment,
  denovo_assembly_name = denovo_assembly_name,
  genome_fasta = genome_fasta,
  genome_guided_assembly_name = genome_guided_assembly_name,
  line = line,
  output_dir = output_dir,
  top_output_dir = top_output_dir,
  threads = threads,
  trimmed_reads_dir = trimmed_reads_dir
)
```

# DE NOVO ASSEMBLY
```{r de-novo-assembly, engine='bash'}
## Inititalize arrays
R1_array=()
R2_array=()

# Variables for R1/R2 lists
R1_list=""
R2_list=""

# Create array of fastq R1 files
R1_array=(${trimmed_reads_dir}/*_R1.fq)

# Create array of fastq R2 files
R2_array=(${trimmed_reads_dir}/*_R2.fq)

# Create list of fastq files used in analysis
## Uses parameter substitution to strip leading path from filename
for fastq in ${reads_dir}/*.fq
do
  echo "${fastq##*/}" >> fastq.list.txt
done

# Create comma-separated lists of FastQ reads
R1_list=$(echo "${R1_array[@]}" | tr " " ",")
R2_list=$(echo "${R2_array[@]}" | tr " " ",")


# Run Trinity using "stranded" setting (--SS_lib_type)
${trinity_dir}/Trinity \
--seqType fq \
--max_memory ${max_ram} \
--CPU ${threads} \
--SS_lib_type RF \
--left "${R1_list}" \
--right "${R2_list}"
```


# Rename generic assembly FastA
mv trinity_out_dir/Trinity.fasta trinity_out_dir/${fasta_name}

# Assembly stats
${trinity_dir}/util/TrinityStats.pl trinity_out_dir/${fasta_name} \
> ${assembly_stats}

# Create gene map files
${trinity_dir}/util/support_scripts/get_Trinity_gene_to_trans_map.pl \
trinity_out_dir/${fasta_name} \
> trinity_out_dir/${fasta_name}.gene_trans_map

# Create FastA index
${samtools} faidx \
trinity_out_dir/${fasta_name}

# GENOME-GUIDED ASSEMBLY
```{r genome-guided-assembly, engine='bash'}
 Trinity --genome_guided_bam ${bam_alignment} \
         --genome_guided_max_intron 10000 \
         --max_memory ${max_ram} \
         --CPU ${threads} 
```


# Rename generic assembly FastA
mv trinity_out_dir/Trinity.fasta trinity_out_dir/${fasta_name}

# Assembly stats
${trinity_dir}/util/TrinityStats.pl trinity_out_dir/${fasta_name} \
> ${assembly_stats}

# Create gene map files
${trinity_dir}/util/support_scripts/get_Trinity_gene_to_trans_map.pl \
trinity_out_dir/${fasta_name} \
> trinity_out_dir/${fasta_name}.gene_trans_map

# Create FastA index
${samtools} faidx \
trinity_out_dir/${fasta_name}