---
title: "07-Apul-miRNA-mRNA-miRanda"
author: "Kathleen Durkin"
date: "2024-12-19"
always_allow_html: true
output: 
  bookdown::html_document2:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
  github_document:
    toc: true
    toc_depth: 3
    number_sections: true
    html_preview: true 
---

miRanda is a target prediction software, used to identify likely miRNA-mRNA interactions.

Inputs:

  - FASTA of A.pulchra 3'UTRs `Apul_3UTR_1kb.fasta`, generated in `05-Apul-annotate-UTRs`
  
  - FASTA of A.pulchra mature miRNAs `miRNA_mature-Apul.fasta`. miRNAs identified in `04-Apul-sRNA-discovery-ShortStack`, matures isolated for use in `06-Apul-miRNA-mRNA-RNAhybrid`

Outputs:






```{r, engine='bash', eval=FALSE}

# score cutoff >100
# energy cutoff <-10
# strict binding

/home/shared/miRanda-3.3a/src/miranda \
../data/06-Apul-miRNA-mRNA-RNAhybrid/miRNA_mature-Apul.fasta \
../output/05-Apul-annotate-UTRs/Apul_3UTR_1kb.fasta \
-sc 100 \
-en -10 \
-strict \
-out ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict_all.tab
```


Let's look at the output

```{r, engine='bash'}

echo "miranda run finished!"
echo "counting number of putative interactions predicted"

zgrep -c "Performing Scan" ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict_all.tab

echo "Parsing output"
grep -A 1 "Scores for this hit:" ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict_all.tab | sort | grep '>' > ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed.txt

echo "counting number of putative interactions predicted"
wc -l ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict_all.tab

```

Let's also annotate each of these putative interactions with the associated gene ID (the gene ID for the mRNA associated with each 3'UTR region input into miRanda). We generated a table associating each 3'UTR region with its parent gene ID in `05-Apul-annotate-UTRs`

```{r}
miRanda_3UTR <- read.table("../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed.txt", header = FALSE, sep = "\t")
geneIDs_3UTR <- read.table("../output/05-Apul-annotate-UTRs/Apul-3UTR-FUNids.txt", header = FALSE, sep = "\t")

geneIDs_3UTR$V4 <- gsub("Parent=", "", geneIDs_3UTR$V4)

# Some genes sit very close to each other and share 3UTR regions (since we manually defined 3UTRs).
# Reformat 3UTR-geneID table to avoid many-to-many problem during left join
geneIDs_3UTR_refmt <- geneIDs_3UTR %>%
  select(V1, V4) %>%
  group_by(V1) %>%
  summarise(V4 = paste(V4, collapse = ","), .groups = "drop")


# left join to annotate 3UTR regions in putative interactions with gene IDs
miRanda_3UTR <- left_join(miRanda_3UTR, geneIDs_3UTR_refmt, by = c("V2" = "V1"))

# save
write.table(miRanda_3UTR, "../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed-geneIDs.txt", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)
```



This is a lot of putative interactions! We can probably narrow it down though. In vertebrates, miRNA-mRNA binding only requires complementarity of an miRNA seed region of ~8 nucleotides. This requirement is built in to miRanda target prediction. In cnidarians, however, miRNA-mRNA binding is believed to require near-complete complementarity of the full mature miRNA, similarly to plants. Let's look at how many putative interactions are predicted for a binding length of at least 21 nucleotides (the length of our smallest mature miRNA).

```{r, engine='bash'}
echo "number of putative interactions of at least 21 nucleotides"
awk -F'\t' '$7 >= 21' ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed.txt | wc -l
echo ""
echo "check some:"
awk -F'\t' '$7 >= 21' ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed.txt | head -5
```

We can also see from the alignment percentages (last 2 entries) that this number includes alignments with multiple mismatches. Let's filter again to reduce the number of permissible mismatches. Let's say we want no more than 3 mismatches. For an alignment of 21 nucleotides, this would be an alignment rate of (21-3)/21 = 85.7%.

```{r, engine='bash'}
echo "number of putative interactions of at least 21 nucleotides, with at most 3 mismatches"
awk -F'\t' '$7 >= 21' ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed.txt | awk -F'\t' '$8 >= 85' | wc -l
echo ""
echo "check some:"
awk -F'\t' '$7 >= 21' ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed.txt | awk -F'\t' '$8 >= 85' | head -5
```
