07.1-Apul-miRNA-mRNA-miRanda-additional_inputs
================
Kathleen Durkin
2024-02-21

- <a href="#1-mrna-coding-sequence" id="toc-1-mrna-coding-sequence">1 mRNA
  coding sequence</a>
  - <a href="#11-get-coding-sequence-fasta"
    id="toc-11-get-coding-sequence-fasta">1.1 Get coding sequence fasta</a>
  - <a href="#12-run-miranda" id="toc-12-run-miranda">1.2 Run miRanda</a>
  - <a href="#13-summarize-results" id="toc-13-summarize-results">1.3
    Summarize results</a>
- <a href="#2-mirna-and-mrna-5utrs" id="toc-2-mirna-and-mrna-5utrs">2
  miRNA and mRNA 5’UTRs</a>
  - <a href="#21-get-5utr-fasta" id="toc-21-get-5utr-fasta">2.1 Get 5’UTR
    fasta</a>
  - <a href="#22-run-miranda" id="toc-22-run-miranda">2.2 Run miRanda</a>
  - <a href="#23-summarize-results" id="toc-23-summarize-results">2.3
    Summarize results</a>
- <a href="#3-mirna-and-mrna-3utrs" id="toc-3-mirna-and-mrna-3utrs">3
  miRNA and mRNA 3’UTRs</a>
- <a href="#4-summary" id="toc-4-summary">4 Summary</a>

miRanda is a target prediction software, used to identify likely
miRNA-mRNA interactions.

We’ve decided that we need to expand our consideration of potential
miRNA target sites to include the coding sequence and (potentially) the
5’UTR region, since cnidarian miRNAs may function primarily through
target cleavage, instead of translational repression.

Run miRanda using the mRNA coding sequences and the 1kb 5UTR regions as
inputs.

# 1 mRNA coding sequence

## 1.1 Get coding sequence fasta

``` bash

/home/shared/bedtools2/bin/bedtools getfasta \
-fi "../data/Apulchra-genome.fa" \
-bed "../output/05-Apul-annotate-UTRs/Apulcra-genome-mRNA_only.gff" \
-fo "../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul_mRNA_full.fa"
```

## 1.2 Run miRanda

``` bash

# score cutoff >100
# energy cutoff <-10
# strict binding

/home/shared/miRanda-3.3a/src/miranda \
../data/06-Apul-miRNA-mRNA-RNAhybrid/miRNA_mature-Apul.fasta \
../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul_mRNA_full.fa \
-sc 100 \
-en -10 \
-strict \
-out ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-mRNA_full-strict_all.tab
```

## 1.3 Summarize results

Let’s look at the output

``` bash

echo "miranda run finished!"
echo "counting number of putative interactions predicted"

zgrep -c "Performing Scan" ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-mRNA_full-strict_all.tab

echo "Parsing output"
grep -A 1 "Scores for this hit:" ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-mRNA_full-strict_all.tab | sort | grep '>' > ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-mRNA_full-strict-parsed.txt

echo "counting number of putative interactions predicted"
wc -l ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-mRNA_full-strict-parsed.txt
```

    ## miranda run finished!
    ## counting number of putative interactions predicted
    ## 1858797
    ## Parsing output
    ## counting number of putative interactions predicted
    ## 814426 ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-mRNA_full-strict-parsed.txt

This is a lot of putative interactions! Note though, that miRanda only
requires complementarity of a 8bp seed region of the miRNA. We instead
want to look for binding with full or near-full complementarity. Let’s
look at how many putative interactions are predicted for a binding
length of at least 21 nucleotides (the length of our smallest mature
miRNA).

``` bash
echo "number of putative interactions of at least 21 nucleotides"
awk -F'\t' '$7 >= 21' ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-mRNA_full-strict-parsed.txt | wc -l
echo ""
echo "check some:"
awk -F'\t' '$7 >= 21' ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-mRNA_full-strict-parsed.txt | head -5
```

    ## number of putative interactions of at least 21 nucleotides
    ## 121397
    ## 
    ## check some:
    ## >Cluster_10452.mature::ptg000020l:10483758-10483779(-)   ntLink_3:108837-109632  151.00  -21.40  2 21    660 684 22  59.09%  77.27%
    ## >Cluster_10452.mature::ptg000020l:10483758-10483779(-)   ntLink_3:128347-130689  151.00  -21.40  2 21    660 684 22  59.09%  77.27%
    ## >Cluster_10452.mature::ptg000020l:10483758-10483779(-)   ntLink_3:50316-56025    151.00  -21.40  2 21    660 684 22  59.09%  77.27%
    ## >Cluster_10452.mature::ptg000020l:10483758-10483779(-)   ntLink_3:69896-75528    151.00  -21.40  2 21    585 609 22  59.09%  77.27%
    ## >Cluster_10452.mature::ptg000020l:10483758-10483779(-)   ntLink_3:89326-95036    151.00  -21.40  2 21    660 684 22  59.09%  77.27%

We can also see from the percent identities (last 2 entries) that this
number includes alignments with multiple mismatches. Let’s filter again
to reduce the number of permissible mismatches. Let’s say we want no
more than 3 mismatches. For an alignment of 21 nucleotides, this would
be an alignment rate of (21-3)/21 = 85.7%.

``` bash
echo "number of putative interactions of at least 21 nucleotides, with at most 3 mismatches"
awk -F'\t' '$7 >= 21' ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-mRNA_full-strict-parsed.txt | awk -F'\t' '$8 >= 85' | wc -l
echo ""
echo "check some:"
awk -F'\t' '$7 >= 21' ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-mRNA_full-strict-parsed.txt | awk -F'\t' '$8 >= 85' | head -5
```

    ## number of putative interactions of at least 21 nucleotides, with at most 3 mismatches
    ## 181
    ## 
    ## check some:
    ## >Cluster_11565.mature::ptg000021l:11127978-11127999(+)   ptg000018l:11470602-11474621    179.00  -20.55  2 21    3778 3801   21  85.71%  85.71%
    ## >Cluster_13327.mature::ptg000023l:18708925-18708946(+)   ntLink_6:11521762-11528244  173.00  -22.70  2 20    1 24    21  85.71%  85.71%
    ## >Cluster_13327.mature::ptg000023l:18708925-18708946(+)   ptg000023l:7445976-7469598  179.00  -23.88  2 21    8990 9013   21  85.71%  85.71%
    ## >Cluster_13647.mature::ptg000023l:37965298-37965318(+)   ptg000023l:25983548-25984865    173.00  -28.51  2 20    591 614 21  85.71%  85.71%
    ## >Cluster_14146.mature::ptg000024l:4086254-4086275(+) ptg000008l:26082763-26089839    183.00  -26.73  2 21    4260 4283   21  85.71%  90.48%

So filtering for full or near-full complementarity reduced the number of
putative interactions between miRNA and full mRNA sequences from 814426
to 181.

# 2 miRNA and mRNA 5’UTRs

We’ve also created a gff of 1kb 5’UTR regions (using the same method
used to define 1kb 3’UTRs, in `05-Apul-annotate-UTRs`). Let’s try
running this through miRanda as well, since it’s possible (though not
necessarily expected) that miRNAs will bind here

## 2.1 Get 5’UTR fasta

``` bash
/home/shared/bedtools2/bin/bedtools getfasta \
-fi "../data/Apulchra-genome.fa" \
-bed "../output/05-Apul-annotate-UTRs/Apul.GFFannotation.5UTR_1kb_corrected.gff" \
-fo "../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul_5UTR_1kb_corrected.fa"
```

## 2.2 Run miRanda

``` bash

# Same settings we've been using: 
# score cutoff >100
# energy cutoff <-10
# strict binding

/home/shared/miRanda-3.3a/src/miranda \
../data/06-Apul-miRNA-mRNA-RNAhybrid/miRNA_mature-Apul.fasta \
../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul_5UTR_1kb_corrected.fa \
-sc 100 \
-en -10 \
-strict \
-out ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-5UTR_1kb-strict_all.tab
```

## 2.3 Summarize results

Let’s look at the output

``` bash

echo "Number of interacting miRNA-lncRNA pairs"
zgrep -c "Performing Scan" ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-5UTR_1kb-strict_all.tab

echo "Parsing output"
grep -A 1 "Scores for this hit:" ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-5UTR_1kb-strict_all.tab | sort | grep '>' > ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-5UTR_1kb-strict-parsed.txt

echo "Number of putative interactions predicted"
wc -l ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-5UTR_1kb-strict-parsed.txt
```

    ## Number of interacting miRNA-lncRNA pairs
    ## 1895058
    ## Parsing output
    ## Number of putative interactions predicted
    ## 115265 ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-5UTR_1kb-strict-parsed.txt

How many putative interactions are predicted for a binding length of at
least 21 nucleotides (the length of our smallest mature miRNA)?

``` bash
echo "number of putative interactions:"
wc -l ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-5UTR_1kb-strict-parsed.txt
echo ""
echo "number of putative interactions of at least 21 nucleotides"
awk -F'\t' '$7 >= 21' ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-5UTR_1kb-strict-parsed.txt | wc -l
echo ""
echo "number of putative interactions of at least 21 nucleotides, with at most 3 mismatches"
awk -F'\t' '$7 >= 21' ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-5UTR_1kb-strict-parsed.txt | awk -F'\t' '$8 >= 85' | wc -l
```

    ## number of putative interactions:
    ## 115265 ../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-5UTR_1kb-strict-parsed.txt
    ## 
    ## number of putative interactions of at least 21 nucleotides
    ## 16779
    ## 
    ## number of putative interactions of at least 21 nucleotides, with at most 3 mismatches
    ## 20

So filtering for full or near-full complementarity reduced the number of
putative interactions between miRNA and 5’UTR sequences from 115265 to
20

# 3 miRNA and mRNA 3’UTRs

Now let’s see how filtering changes the outputs of miRanda run with only
the 3’UTR mRNA region (the input we have been using up till now)

``` bash
echo "total number of putative interactions:"
wc -l ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed.txt

echo "number of putative interactions of at least 21 nucleotides"
awk -F'\t' '$7 >= 21' ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed.txt | wc -l

echo "number of putative interactions of at least 21 nucleotides, with at most 3 mismatches"
awk -F'\t' '$7 >= 21' ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed.txt | awk -F'\t' '$8 >= 85' | wc -l
```

    ## total number of putative interactions:
    ## 123123 ../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed.txt
    ## number of putative interactions of at least 21 nucleotides
    ## 18420
    ## number of putative interactions of at least 21 nucleotides, with at most 3 mismatches
    ## 33

When only mRNA 3’UTR regions are used as input, filtering for
full/near-full complementarity reduces the number of putative
interactions from 123123 to 33.

# 4 Summary

| Input     | unfiltered | filtered for complementarity | % retained |
|:----------|:-----------|:-----------------------------|:-----------|
| full mRNA | 814426     | 181                          | 0.0222 %   |
| 5’UTR     | 115265     | 20                           | 0.0174 %   |
| 3’UTR     | 123123     | 33                           | 0.0268 %   |
