---
title: "P.tuahiniensis GFF reformatting"
author: "Sam White"
date: "2025-01-14"
output: 
  bookdown::html_document2:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
  github_document:
    toc: true
    number_sections: true
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
---

# INTRO

This notebook reformats the original _P.tuahiniensis_ GFF ([Pocillopora_meandrina_HIv1.genes.gff3](../data/Pocillopora_meandrina_HIv1.genes.gff3)),

Finally, despite the naming convention, there aren't any actual annotations in that GFF, beyond the feature designations (i.e. no gene ontology, no SwissProt IDs, gene names, etc.). This notebook does _not_ address those shortcomings.

::: {.callout-note}
Unlike other scripts, this will output to [`F-Ptua/data`](../data), instead of an output directory in `../output`.
:::

## Software requirements

Requires [genometools](https://github.com/genometools/genometools) (GitHub repo) to be installed and in the system `$PATH`.

## Inputs

- [Pocillopora_meandrina_HIv1.genes.gff3](../data/Pocillopora_meandrina_HIv1.genes.gff3)

## Outputs

- [](../data/)

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  comment = ""         # Prevents appending '##' to beginning of lines in code output
)
```

# Load libraries
```{r load-libraries}
library(tidyverse)
```

# Read GFF3 file
```{r load-input-gff}
gff <- read_delim("../data/Pocillopora_meandrina_HIv1.genes.gff3", 
                  delim="\t", 
                  col_names=c("seqid", "source", "type", "start", "end", 
                            "score", "strand", "phase", "attributes"),
                  show_col_types = FALSE)
str(gff)
```

# Get unique types
```{r unqique-features, eval=TRUE}
# Get and print unique types, one per line
cat("Unique feature types in original GFF3:\n")
gff %>%
  distinct(type) %>%
  pull(type) %>%
  walk(~cat("-", .x, "\n"))
```

# Create gene entries from transcript entries
```{r create-gene-entries}
# Create gene entries from transcripts
gene_entries <- gff %>%
  filter(type == "transcript") %>%
  mutate(
    type = "gene",
    # Extract base gene ID (g23774)
    base_id = str_extract(attributes, "g\\d+"),
    # Create new gene ID format
    attributes = paste0("ID=gene-", base_id)
  ) %>%
  select(-base_id)
str(gene_entries)
```

# Create mRNA, Feature IDs, and Parent Attributes
```{r replace-transcript-with-mRNA}
modified_gff <- gff %>%
  mutate(
    # Convert transcript to mRNA
    type = if_else(type == "transcript", "mRNA", type),
    # Create base ID for each feature
    base_id = str_extract(attributes, "g\\d+\\.t\\d+"),
    # Create lowercase type for prefixes
    type_lower = tolower(type),
    # Determine parent type
    parent_type = case_when(
      type == "mRNA" ~ "gene",
      TRUE ~ "mrna"
    ),
    # Build new attributes string
    attributes = case_when(
      type == "gene" ~ paste0("ID=", type_lower, "-", base_id),
      TRUE ~ paste0("ID=", type_lower, "-", base_id, 
                   ";Parent=", parent_type, "-", base_id)
    )
  ) %>%
  select(-base_id, -type_lower, -parent_type)

str(modified_gff)
```

# Number exons within mRNA groups
```{r number-exons}
exons_gff <- modified_gff %>%
  # Group by mRNA boundaries
  group_by(seqid, 
           mRNA_group = cumsum(type == "mRNA")) %>%
  # Add exon numbers within groups
  mutate(
    # Only count when type is exon
    exon_num = if_else(type == "exon",
                       cumsum(type == "exon"),
                       NA_integer_),
    # Modify attributes to append exon number
    attributes = if_else(
      type == "exon",
      str_replace(attributes, 
                 "(ID=exon-[^;]+)",
                 paste0("\\1-", exon_num)),
      attributes
    )
  ) %>%
  ungroup() %>%
  select(-mRNA_group, -exon_num) %>%
  arrange(seqid, start)
```

# Combine and sort
```{r combine-and-sort}
final_gff <- bind_rows(gene_entries, exons_gff) %>%
  arrange(seqid, start)

str(final_gff)
```

# Unique features
```{r unqique-features-updated-gff, eval=TRUE}
# Get and print unique types, one per line
cat("Unique feature types in final GFF3:\n")
final_gff %>%
  distinct(type) %>%
  pull(type) %>%
  walk(~cat("-", .x, "\n"))
```

# Write output
```{r write-reformatted-gff}
write_delim(final_gff, 
            "Pocillopora_meandrina_HIv1.genes.modified.gff3", 
            delim="\t",
            col_names=FALSE)
```