00.20-F-Ptua-WGBS-reads-FastQC-MultiQC
================
Sam White
2025-02-10

- <a href="#1-create-a-bash-variables-file"
  id="toc-1-create-a-bash-variables-file">1 Create a Bash variables
  file</a>
- <a href="#2-download-ptuahiniensis-wgbs-fastqs"
  id="toc-2-download-ptuahiniensis-wgbs-fastqs">2 Download
  <em>P.tuahiniensis</em> WGBS FastQs</a>
  - <a href="#21-inspect-metadata-file"
    id="toc-21-inspect-metadata-file">2.1 Inspect metadata file</a>
  - <a href="#22-download-raw-wgbs-reads"
    id="toc-22-download-raw-wgbs-reads">2.2 Download raw WGBS reads</a>
  - <a
    href="#23-compare-counts-of-expected-sample-files-to-downloaded-files"
    id="toc-23-compare-counts-of-expected-sample-files-to-downloaded-files">2.3
    Compare counts of expected sample files to downloaded files</a>
  - <a href="#24-verify-raw-read-checksums"
    id="toc-24-verify-raw-read-checksums">2.4 Verify raw read checksums</a>
- <a href="#3-rename-fastqs-for-project-consistency"
  id="toc-3-rename-fastqs-for-project-consistency">3 Rename FastQs for
  Project Consistency</a>
  - <a href="#31-count-renamed-files" id="toc-31-count-renamed-files">3.1
    Count renamed files</a>
  - <a href="#32-create-md5-files-for-new-filenames"
    id="toc-32-create-md5-files-for-new-filenames">3.2 Create MD5 files for
    new filenames</a>
- <a href="#4-fastqcmultiqc-on-raw-reads"
  id="toc-4-fastqcmultiqc-on-raw-reads">4 FastQC/MultiQC on raw reads</a>

This Rmd file will download raw WGBS FastQs for *P.tuahiniensis* and
evaluate them using [FastQC](https://github.com/s-andrews/FastQC) and
[MultiQC](https://multiqc.info/)(Ewels et al. 2016).

# 1 Create a Bash variables file

This allows usage of Bash variables across R Markdown chunks.

``` bash
{
echo "#### Assign Variables ####"
echo ""

echo "# Data directories"
echo 'export timeseries_dir=/home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular'
echo 'export output_dir_top=${timeseries_dir}/F-Ptua/output/00.20-F-Ptua-WGBS-reads-FastQC-MultiQC'
echo 'export raw_reads_dir=${timeseries_dir}/F-Ptua/data/wgbs-raw-fastqs'
echo 'export raw_reads_url="https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1067895835/"'
echo ""

echo "# Input files"
echo 'export metadata_file="${timeseries_dir}/M-multi-species/data/e5_DNA_Azenta_metadata.csv"'

echo "# Paths to programs"
echo 'export fastqc=/home/shared/FastQC-0.12.1/fastqc'
echo 'export multiqc=/home/sam/programs/mambaforge/bin/multiqc'
echo ""

echo "# Set FastQ filename patterns"
echo "export fastq_pattern='*.fastq.gz'"
echo "export R1_fastq_pattern='*_R1_*.fastq.gz'"
echo "export R2_fastq_pattern='*_R2_*.fastq.gz'"
echo ""

echo "# Set number of CPUs to use"
echo 'export threads=40'
echo ""


echo "## Inititalize arrays"
echo 'export fastq_array_R1=()'
echo 'export fastq_array_R2=()'
echo 'export raw_fastqs_array=()'
echo 'export R1_names_array=()'
echo 'export R2_names_array=()'
echo ""

echo "# Programs associative array"
echo "declare -A programs_array"
echo "programs_array=("
echo '[fastqc]="${fastqc}" \'
echo '[multiqc]="${multiqc}" \'
echo ")"
echo ""

echo "# Print formatting"
echo 'export line="--------------------------------------------------------"'
echo ""
} > .bashvars

cat .bashvars
```

    #### Assign Variables ####

    # Data directories
    export timeseries_dir=/home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular
    export output_dir_top=${timeseries_dir}/F-Ptua/output/00.20-F-Ptua-WGBS-reads-FastQC-MultiQC
    export raw_reads_dir=${timeseries_dir}/F-Ptua/data/wgbs-raw-fastqs
    export raw_reads_url="https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1067895835/"

    # Input files
    export metadata_file="${timeseries_dir}/M-multi-species/data/e5_DNA_Azenta_metadata.csv"
    # Paths to programs
    export fastqc=/home/shared/FastQC-0.12.1/fastqc
    export multiqc=/home/sam/programs/mambaforge/bin/multiqc

    # Set FastQ filename patterns
    export fastq_pattern='*.fastq.gz'
    export R1_fastq_pattern='*_R1_*.fastq.gz'
    export R2_fastq_pattern='*_R2_*.fastq.gz'

    # Set number of CPUs to use
    export threads=40

    ## Inititalize arrays
    export fastq_array_R1=()
    export fastq_array_R2=()
    export raw_fastqs_array=()
    export R1_names_array=()
    export R2_names_array=()

    # Programs associative array
    declare -A programs_array
    programs_array=(
    [fastqc]="${fastqc}" \
    [multiqc]="${multiqc}" \
    )

    # Print formatting
    export line="--------------------------------------------------------"

# 2 Download *P.tuahiniensis* WGBS FastQs

## 2.1 Inspect metadata file

``` bash
# Load bash variables into memory
source .bashvars

head ${metadata_file} |
column -t -s","
```

    Sample Name*  Plate         Well number  Azenta sample name  Colony ID  Timepoint  Sample Type*  Species/Strain*  Sample Buffer*    Total Amount (ng)*    Volume (µL)  Conc (ng/µL)  Purification Method*  Biosafety Level (BSL)            Method Used for Fluorescence  Method Used for Spectrophotometry  Method Used for Electrophoresis  
    1             41            1            G4                  1G4        ACR-225    TP1           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  2064         80            25.8                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
    2             291-20220211  2            A3                  2A3        ACR-225    TP2           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  5064         80            63.3                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
    3             527           1            H5                  1H5        ACR-225    TP3           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  4232         80            52.9                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
    4             911           1            C6                  1C6        ACR-225    TP4           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  4488         80            56.1                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
    5             55            1            B2                  1B2        ACR-229    TP1           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  2432         80            30.4                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
    6             295           1            E3                  1E3        ACR-229    TP2           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  2760         80            34.5                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
    7             531           1            H3                  1H3        ACR-229    TP3           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  5752         80            71.9                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
    8             931           1            F7                  1F7        ACR-229    TP4           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  6008         80            75.1                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
    9             49            1            A1                  1A1        ACR-237    TP1           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  1456         80            18.2                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"

## 2.2 Download raw WGBS reads

Reads are downloaded from
<https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1067895835/>

Since sequencing included multiple species, the code will also parse
only those that are *P.tuahiniensis*.

The `--cut-dirs 3` command cuts the preceding directory structure
(i.e. `nightingales/E5-coral-time-series/30-1067895835/`) so that we
just end up with the reads.

``` bash
# Load bash variables into memory
source .bashvars

# Make output directory if it doesn't exist
mkdir --parents ${raw_reads_dir}

# Create list of only P.tuahiniensis sample names
sample_list=$(awk -F"," '$9 == "Pocillopora tuahiniensis" { print $5 }' ${timeseries_dir}/M-multi-species/data/e5_DNA_Azenta_metadata.csv \
| sort)

echo ""
echo "${line}"
echo ""
echo "Sample list:"
echo ""
echo "${sample_list}"
echo ""
echo "${line}"
echo ""


# Use printf to format each item for use in wget
formatted_list=$(printf "*%s_*," ${sample_list})

# Remove the trailing comma
formatted_list="${formatted_list%,}"

# Output the final wget command
echo ""
echo "${line}"
echo ""
echo "Formatted wget accept list:"
echo ""
echo "wget --accept=\"$formatted_list\""
echo ""
echo "${line}"
echo ""
# Set a log file
wget_log="${raw_reads_dir}/wget_log.txt"

# Run wget to retrieve FastQs and MD5 files
wget \
--directory-prefix ${raw_reads_dir} \
--recursive \
--no-check-certificate \
--continue \
--cut-dirs 3 \
--no-host-directories \
--no-parent \
--quiet \
--accept=${formatted_list} \
${raw_reads_url}

# List downloaded files
downloaded_files=$(find "${raw_reads_dir}" -maxdepth 1 -type f -name "*.fastq.gz" -print)

echo ""
echo "${line}"
echo ""
echo "Downloaded files:"
echo "$downloaded_files"
echo ""
echo "${line}"
echo ""

ls -lh "${raw_reads_dir}"
```

    --------------------------------------------------------

    Sample list:

    1A11
    1A2
    1A4
    1B12
    1B7
    1C1
    1C10
    1C11
    1C12
    1C3
    1C4
    1C5
    1C7
    1D4
    1D5
    1E1
    1E10
    1E12
    1E2
    1F11
    1F12
    1F2
    1F4
    1F8
    1G1
    1G12
    1G3
    1G7
    1G8
    1H1
    1H10
    1H4
    1H7
    2A1
    2B2
    2D3
    2E3
    2F1
    2H1

    --------------------------------------------------------


    --------------------------------------------------------

    Formatted wget accept list:

    wget --accept="*1A11_*,*1A2_*,*1A4_*,*1B12_*,*1B7_*,*1C1_*,*1C10_*,*1C11_*,*1C12_*,*1C3_*,*1C4_*,*1C5_*,*1C7_*,*1D4_*,*1D5_*,*1E1_*,*1E10_*,*1E12_*,*1E2_*,*1F11_*,*1F12_*,*1F2_*,*1F4_*,*1F8_*,*1G1_*,*1G12_*,*1G3_*,*1G7_*,*1G8_*,*1H1_*,*1H10_*,*1H4_*,*1H7_*,*2A1_*,*2B2_*,*2D3_*,*2E3_*,*2F1_*,*2H1_*"

    --------------------------------------------------------


    --------------------------------------------------------

    Downloaded files:
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1F2_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/2A1_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/66--1E2_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1G3_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1H10_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C4_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/2D3_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1B12_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1F2_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/58--1E10_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C3_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1B12_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1F11_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1D5_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1D4_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/2A1_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C7_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/2B2_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C7_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/55--2E3_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C3_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C12_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C5_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1A11_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1A4_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1F4_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1H10_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1A2_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1G12_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1H1_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1B7_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1H7_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C1_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1F8_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C10_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C11_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1G12_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/59--1E1_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/59--1E1_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1F12_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/2F1_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1F4_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1D5_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/66--1E2_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1A4_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1F8_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1A11_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C10_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/2H1_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1H4_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1G1_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1A2_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1B7_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C11_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1G8_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1G8_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/49--1E12_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C12_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/2D3_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1G7_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1H4_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/2B2_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1H1_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/49--1E12_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1H7_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1G3_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1F12_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/55--2E3_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/2H1_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C1_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C5_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1C4_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/2F1_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1G7_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/58--1E10_R2_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1D4_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1F11_R1_001.fastq.gz
    /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs/1G1_R2_001.fastq.gz

    --------------------------------------------------------

    total 202G
    -rw-r--r-- 1 sam sam 2.8G Dec 30 08:27 1A11_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 08:27 1A11_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.8G Dec 30 08:30 1A11_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 08:30 1A11_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 3.0G Dec 30 08:44 1A2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 08:44 1A2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 3.1G Dec 30 08:48 1A2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 08:48 1A2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 3.0G Dec 30 08:57 1A4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 08:57 1A4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.9G Dec 30 08:59 1A4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 08:59 1A4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.7G Dec 30 09:42 1B12_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 09:42 1B12_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.7G Dec 30 09:45 1B12_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 09:45 1B12_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 10:21 1B7_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 10:21 1B7_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 10:23 1B7_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 10:23 1B7_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.8G Dec 30 10:34 1C10_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 10:34 1C10_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.8G Dec 30 10:37 1C10_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 10:37 1C10_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.8G Dec 30 10:40 1C11_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 10:40 1C11_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.8G Dec 30 10:44 1C11_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 10:44 1C11_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.3G Dec 30 10:46 1C12_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 10:46 1C12_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.2G Dec 30 10:48 1C12_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 10:48 1C12_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 3.3G Dec 30 10:51 1C1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 10:51 1C1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 3.3G Dec 30 10:54 1C1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 10:54 1C1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 11:01 1C3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 11:01 1C3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 11:04 1C3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 11:04 1C3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.0G Dec 30 11:06 1C4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 11:06 1C4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.0G Dec 30 11:08 1C4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 11:08 1C4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.2G Dec 30 11:10 1C5_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 11:10 1C5_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.2G Dec 30 11:13 1C5_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 11:13 1C5_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.8G Dec 30 11:20 1C7_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 11:20 1C7_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.8G Dec 30 11:23 1C7_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 11:23 1C7_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 3.1G Dec 30 12:07 1D4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 12:07 1D4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 3.1G Dec 30 12:10 1D4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 12:10 1D4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.2G Dec 30 12:12 1D5_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 12:12 1D5_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.2G Dec 30 12:14 1D5_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 12:14 1D5_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.9G Dec 30 12:41 1F11_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 12:41 1F11_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.9G Dec 30 12:44 1F11_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 12:44 1F11_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.4G Dec 30 12:46 1F12_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 12:46 1F12_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.4G Dec 30 12:48 1F12_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 12:48 1F12_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 12:56 1F2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 12:56 1F2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 12:58 1F2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 12:58 1F2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.5G Dec 30 13:04 1F4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 13:04 1F4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 13:07 1F4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 13:07 1F4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.5G Dec 30 13:23 1F8_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 13:23 1F8_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.5G Dec 30 13:26 1F8_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 13:26 1F8_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.2G Dec 30 13:41 1G12_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 13:41 1G12_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.2G Dec 30 13:44 1G12_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 13:44 1G12_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.8G Dec 30 13:46 1G1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 13:46 1G1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.7G Dec 30 13:49 1G1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 13:49 1G1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.7G Dec 30 13:57 1G3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 13:57 1G3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.7G Dec 30 13:59 1G3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 13:59 1G3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.9G Dec 30 14:14 1G7_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 14:14 1G7_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.8G Dec 30 14:17 1G7_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 14:17 1G7_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.7G Dec 30 14:20 1G8_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 14:20 1G8_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.7G Dec 30 14:22 1G8_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 14:22 1G8_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 3.3G Dec 30 14:30 1H10_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 14:30 1H10_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 3.3G Dec 30 14:33 1H10_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   57 Dec 30 14:33 1H10_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.5G Dec 30 14:46 1H1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 14:46 1H1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.5G Dec 30 14:48 1H1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 14:48 1H1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 15:01 1H4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 15:01 1H4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 15:04 1H4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 15:04 1H4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.3G Dec 30 15:19 1H7_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 15:19 1H7_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.2G Dec 30 15:22 1H7_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 15:22 1H7_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.4G Dec 30 15:36 2A1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 15:36 2A1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.4G Dec 30 15:38 2A1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 15:38 2A1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.9G Dec 30 15:54 2B2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 15:54 2B2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.9G Dec 30 15:57 2B2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 15:57 2B2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 1.8G Dec 30 16:32 2D3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 16:32 2D3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 1.8G Dec 30 16:34 2D3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 16:34 2D3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 16:37 2F1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 16:37 2F1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.7G Dec 30 16:40 2F1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 16:40 2F1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.7G Dec 30 17:10 2H1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 17:10 2H1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.8G Dec 30 17:14 2H1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   56 Dec 30 17:14 2H1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.2G Dec 30 17:32 49--1E12_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 Dec 30 17:32 49--1E12_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.2G Dec 30 17:35 49--1E12_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 Dec 30 17:35 49--1E12_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.5G Dec 30 17:38 55--2E3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   60 Dec 30 17:38 55--2E3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.5G Dec 30 17:41 55--2E3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   60 Dec 30 17:41 55--2E3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 3.0G Dec 30 17:44 58--1E10_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 Dec 30 17:44 58--1E10_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 3.0G Dec 30 17:47 58--1E10_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 Dec 30 17:47 58--1E10_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.9G Dec 30 17:50 59--1E1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   60 Dec 30 17:50 59--1E1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.9G Dec 30 17:54 59--1E1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   60 Dec 30 17:54 59--1E1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 18:03 66--1E2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   60 Dec 30 18:03 66--1E2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 2.6G Dec 30 18:06 66--1E2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   60 Dec 30 18:06 66--1E2_R2_001.fastq.gz.md5

## 2.3 Compare counts of expected sample files to downloaded files

``` bash
# Load bash variables into memory
source .bashvars

# Count expected number of P.tuahiniensis samples (assuming each has an R1 file)
expected_count=$(awk -F"," '$9 == "Pocillopora tuahiniensis" { count++ } END {print count}' "${metadata_file}")

# Count the number of R1 fastq files in the raw reads directory
actual_count=$(find "${raw_reads_dir}" -maxdepth 1 -type f -name "*_R1_*.fastq.gz" | wc -l)

echo ""
echo "${line}"
echo ""
echo "Expected number of R1 files: ${expected_count}"
echo "Actual number of R1 files: ${actual_count}"
echo ""
echo "${line}"
echo ""

# Compare the counts
if [ "$expected_count" -eq "$actual_count" ]; then
  echo "PASS: All expected R1 files downloaded."
else
  echo "FAIL: Mismatch in expected and actual R1 file counts."
fi
echo ""
echo "${line}"
echo ""
```

    --------------------------------------------------------

    Expected number of R1 files: 39
    Actual number of R1 files: 39

    --------------------------------------------------------

    PASS: All expected R1 files downloaded.

    --------------------------------------------------------

## 2.4 Verify raw read checksums

``` bash
# Load bash variables into memory
source .bashvars

cd "${raw_reads_dir}"

for file in *.md5
do
  md5sum --check "${file}"
done
```

    ./1A11_R1_001.fastq.gz: OK
    ./1A11_R2_001.fastq.gz: OK
    ./1A2_R1_001.fastq.gz: OK
    ./1A2_R2_001.fastq.gz: OK
    ./1A4_R1_001.fastq.gz: OK
    ./1A4_R2_001.fastq.gz: OK
    ./1B12_R1_001.fastq.gz: OK
    ./1B12_R2_001.fastq.gz: OK
    ./1B7_R1_001.fastq.gz: OK
    ./1B7_R2_001.fastq.gz: OK
    ./1C10_R1_001.fastq.gz: OK
    ./1C10_R2_001.fastq.gz: OK
    ./1C11_R1_001.fastq.gz: OK
    ./1C11_R2_001.fastq.gz: OK
    ./1C12_R1_001.fastq.gz: OK
    ./1C12_R2_001.fastq.gz: OK
    ./1C1_R1_001.fastq.gz: OK
    ./1C1_R2_001.fastq.gz: OK
    ./1C3_R1_001.fastq.gz: OK
    ./1C3_R2_001.fastq.gz: OK
    ./1C4_R1_001.fastq.gz: OK
    ./1C4_R2_001.fastq.gz: OK
    ./1C5_R1_001.fastq.gz: OK
    ./1C5_R2_001.fastq.gz: OK
    ./1C7_R1_001.fastq.gz: OK
    ./1C7_R2_001.fastq.gz: OK
    ./1D4_R1_001.fastq.gz: OK
    ./1D4_R2_001.fastq.gz: OK
    ./1D5_R1_001.fastq.gz: OK
    ./1D5_R2_001.fastq.gz: OK
    ./1F11_R1_001.fastq.gz: OK
    ./1F11_R2_001.fastq.gz: OK
    ./1F12_R1_001.fastq.gz: OK
    ./1F12_R2_001.fastq.gz: OK
    ./1F2_R1_001.fastq.gz: OK
    ./1F2_R2_001.fastq.gz: OK
    ./1F4_R1_001.fastq.gz: OK
    ./1F4_R2_001.fastq.gz: OK
    ./1F8_R1_001.fastq.gz: OK
    ./1F8_R2_001.fastq.gz: OK
    ./1G12_R1_001.fastq.gz: OK
    ./1G12_R2_001.fastq.gz: OK
    ./1G1_R1_001.fastq.gz: OK
    ./1G1_R2_001.fastq.gz: OK
    ./1G3_R1_001.fastq.gz: OK
    ./1G3_R2_001.fastq.gz: OK
    ./1G7_R1_001.fastq.gz: OK
    ./1G7_R2_001.fastq.gz: OK
    ./1G8_R1_001.fastq.gz: OK
    ./1G8_R2_001.fastq.gz: OK
    ./1H10_R1_001.fastq.gz: OK
    ./1H10_R2_001.fastq.gz: OK
    ./1H1_R1_001.fastq.gz: OK
    ./1H1_R2_001.fastq.gz: OK
    ./1H4_R1_001.fastq.gz: OK
    ./1H4_R2_001.fastq.gz: OK
    ./1H7_R1_001.fastq.gz: OK
    ./1H7_R2_001.fastq.gz: OK
    ./2A1_R1_001.fastq.gz: OK
    ./2A1_R2_001.fastq.gz: OK
    ./2B2_R1_001.fastq.gz: OK
    ./2B2_R2_001.fastq.gz: OK
    ./2D3_R1_001.fastq.gz: OK
    ./2D3_R2_001.fastq.gz: OK
    ./2F1_R1_001.fastq.gz: OK
    ./2F1_R2_001.fastq.gz: OK
    ./2H1_R1_001.fastq.gz: OK
    ./2H1_R2_001.fastq.gz: OK
    ./49--1E12_R1_001.fastq.gz: OK
    ./49--1E12_R2_001.fastq.gz: OK
    ./55--2E3_R1_001.fastq.gz: OK
    ./55--2E3_R2_001.fastq.gz: OK
    ./58--1E10_R1_001.fastq.gz: OK
    ./58--1E10_R2_001.fastq.gz: OK
    ./59--1E1_R1_001.fastq.gz: OK
    ./59--1E1_R2_001.fastq.gz: OK
    ./66--1E2_R1_001.fastq.gz: OK
    ./66--1E2_R2_001.fastq.gz: OK

# 3 Rename FastQs for Project Consistency

New naming format will be `<colony_id>-<timepoint>`.

Use of a temp file, instead of working directory from the metadata file
can avoid potential issues with character/text formatting from the input
CSV.

``` bash
# Load bash variables into memory
source .bashvars

cd "${raw_reads_dir}"

# Create an associative array to store the mapping
declare -A number_to_colony_timepoint

# Create a temporary file to store the mapping
temp_mapping_file=$(mktemp)

# Read the metadata file and populate the temporary file using awk
# Maps Azenta sample name to colony_id and timepoint.
awk -F, 'NR > 1 { print $5, $6 "-" $7 }' "$metadata_file" > "$temp_mapping_file"

# Read the temporary file and populate the associative array
while read -r Number colony_timepoint; do
    number_to_colony_timepoint["$Number"]="$colony_timepoint"
done < "$temp_mapping_file"

# Remove the temporary file
rm "$temp_mapping_file"

# Print the associative array for debugging
echo "Number to colony_id-Timepoint mapping:"
for key in "${!number_to_colony_timepoint[@]}"; do
    echo ""
    echo "$key -> ${number_to_colony_timepoint[$key]}"
    echo ""
done

# Iterate over the FastQ files in the current directory
for fastq_file in *.fastq.gz; do
    # Extract the sample number from the filename
    sample_number=$(echo "$fastq_file" | sed -E 's/^([0-9]+--)?([A-Za-z0-9]+).*/\2/')
    
    # Check if the sample number exists in the associative array
    if [[ -n "${number_to_colony_timepoint[$sample_number]}" ]]; then
        new_sample_name="${number_to_colony_timepoint[$sample_number]}"
        new_filename=$(echo "$fastq_file" | sed -E "s/^([0-9]+--)?$sample_number/$new_sample_name/")
        if [[ "$fastq_file" != "$new_filename" ]]; then
            mv "$fastq_file" "$new_filename"
            echo "Renamed $fastq_file to $new_filename"
        else
            echo "No renaming needed for $fastq_file"
        fi
    else
        echo "Sample number '$sample_number' not found in metadata."
        # Debugging: Print all keys in the associative array with quotes
        echo "Available keys in the associative array:"
        for key in "${!number_to_colony_timepoint[@]}"; do
            echo "'$key'"
        done
    fi
done
```

    Number to colony_id-Timepoint mapping:

    1B3 -> SR_1-440-NA


    1B2 -> ACR-229-TP1


    1B1 -> ACR-150-TP3


    1B7 -> POC-40-TP3


    1B6 -> POR-245-TP2


    1B5 -> ACR-244-TP2


    1B4 -> ACR-244-TP4


    1B9 -> POR-236-TP2


    1B8 -> POR-69-TP1


    1D5 -> POC-57-TP1


    1D4 -> POC-52-TP3


    1C1 -> POC-222-TP3


    1C2 -> POR-83-TP4


    1D7 -> ACR-186-TP4


    1D6 -> POR-245-TP4


    1C3 -> POC-53-TP3


    1D1 -> POR-216-TP2


    1B12 -> POC-219-TP4


    1C4 -> POC-201-TP2


    1C5 -> POC-52-TP4


    1B10 -> ACR-150-TP1


    1D3 -> ACR-237-TP4


    1C6 -> ACR-225-TP4


    1B11 -> POR-69-TP4


    1D2 -> POR-260-TP1


    1C7 -> POC-53-TP1


    1C8 -> ACR-145-TP2


    1C9 -> ACR-237-TP3


    2C3 -> POR-73-TP1


    1D9 -> ACR-139-TP3


    2C2 -> ACR-145-TP3


    1D8 -> ACR-139-TP4


    2C1 -> POR-216-TP1


    1E2 -> POC-42-TP3


    1E3 -> ACR-229-TP2


    1E1 -> POC-259-TP4


    1E6 -> POR-83-TP1


    1E7 -> ACR-237-TP2


    1E4 -> POR-74-TP2


    1E5 -> POR-245-TP3


    1E11 -> POR-260-TP4


    1E10 -> POC-259-TP3


    1E8 -> POR-262-TP3


    1E9 -> POR-73-TP4


    1E12 -> POC-222-TP2


    2B1 -> SR_3-440B-NA


    2B2 -> POC-40-TP2


    2B3 -> POR-262-TP4


    2E1 -> POR-74-TP1


    2E3 -> POC-255-TP4


    2E2 -> ACR-186-TP3


    1H1 -> POC-57-TP4


    1H3 -> ACR-229-TP3


    1H2 -> ACR-173-TP3


    2D2 -> ACR-173-TP2


    1H5 -> ACR-225-TP3


    2D3 -> POC-52-TP1


    1H4 -> POC-201-TP1


    1H7 -> POC-219-TP1


    1F9 -> ACR-265-TP2


    1F8 -> POC-53-TP4


    2D1 -> ACR-150-TP4


    1H6 -> ACR-150-TP2


    1H9 -> ACR-173-TP1


    1F7 -> ACR-229-TP4


    1H8 -> POR-74-TP4


    1F6 -> POR-260-TP2


    1F5 -> ACR-145-TP4


    1F4 -> POC-255-TP1


    1F3 -> POR-83-TP3


    1D10 -> POR-69-TP2


    1D11 -> SR_2-84-NA


    1F2 -> POC-57-TP3


    1D12 -> POR-73-TP3


    1F1 -> ACR-244-TP1


    2G3 -> POR-74-TP3


    2G2 -> ACR-265-TP1


    2G1 -> POR-245-TP1


    1G8 -> POC-222-TP1


    1G9 -> POR-260-TP3


    1G12 -> POC-40-TP1


    1G11 -> POR-216-TP3


    1G10 -> POR-72-TP3


    1G4 -> ACR-225-TP1


    1G5 -> ACR-173-TP4


    1G6 -> POR-236-TP1


    1G7 -> POC-222-TP4


    1G1 -> POC-57-TP2


    1G2 -> POR-216-TP4


    1G3 -> POC-259-TP1


    2F1 -> POC-255-TP3


    2F2 -> POR-83-TP2


    2F3 -> POR-72-TP4


    2A1 -> POC-40-TP4


    1A11 -> POC-201-TP3


    1A10 -> ACR-139-TP1


    2A3 -> ACR-225-TP2


    2A2 -> POR-72-TP2


    1A12 -> ACR-186-TP2


    2H2 -> POR-262-TP2


    2H3 -> ACR-244-TP3


    2H1 -> POC-42-TP1


    1H10 -> POC-219-TP2


    1A6 -> ACR-265-TP3


    1A7 -> POR-262-TP1


    1C12 -> POC-52-TP2


    1H11 -> ACR-265-TP4


    1H12 -> POR-73-TP2


    1C11 -> POC-42-TP4


    1A4 -> POC-259-TP2


    1C10 -> POC-53-TP2


    1A5 -> ACR-139-TP2


    1A2 -> POC-219-TP3


    1A3 -> ACR-145-TP1


    1A1 -> ACR-237-TP1


    1F12 -> POC-255-TP2


    1F10 -> POR-72-TP1


    1F11 -> POC-42-TP2


    1A8 -> ACR-186-TP1


    1A9 -> POR-69-TP3

    Renamed 1A11_R1_001.fastq.gz to POC-201-TP3_R1_001.fastq.gz
    Renamed 1A11_R2_001.fastq.gz to POC-201-TP3_R2_001.fastq.gz
    Renamed 1A2_R1_001.fastq.gz to POC-219-TP3_R1_001.fastq.gz
    Renamed 1A2_R2_001.fastq.gz to POC-219-TP3_R2_001.fastq.gz
    Renamed 1A4_R1_001.fastq.gz to POC-259-TP2_R1_001.fastq.gz
    Renamed 1A4_R2_001.fastq.gz to POC-259-TP2_R2_001.fastq.gz
    Renamed 1B12_R1_001.fastq.gz to POC-219-TP4_R1_001.fastq.gz
    Renamed 1B12_R2_001.fastq.gz to POC-219-TP4_R2_001.fastq.gz
    Renamed 1B7_R1_001.fastq.gz to POC-40-TP3_R1_001.fastq.gz
    Renamed 1B7_R2_001.fastq.gz to POC-40-TP3_R2_001.fastq.gz
    Renamed 1C10_R1_001.fastq.gz to POC-53-TP2_R1_001.fastq.gz
    Renamed 1C10_R2_001.fastq.gz to POC-53-TP2_R2_001.fastq.gz
    Renamed 1C11_R1_001.fastq.gz to POC-42-TP4_R1_001.fastq.gz
    Renamed 1C11_R2_001.fastq.gz to POC-42-TP4_R2_001.fastq.gz
    Renamed 1C12_R1_001.fastq.gz to POC-52-TP2_R1_001.fastq.gz
    Renamed 1C12_R2_001.fastq.gz to POC-52-TP2_R2_001.fastq.gz
    Renamed 1C1_R1_001.fastq.gz to POC-222-TP3_R1_001.fastq.gz
    Renamed 1C1_R2_001.fastq.gz to POC-222-TP3_R2_001.fastq.gz
    Renamed 1C3_R1_001.fastq.gz to POC-53-TP3_R1_001.fastq.gz
    Renamed 1C3_R2_001.fastq.gz to POC-53-TP3_R2_001.fastq.gz
    Renamed 1C4_R1_001.fastq.gz to POC-201-TP2_R1_001.fastq.gz
    Renamed 1C4_R2_001.fastq.gz to POC-201-TP2_R2_001.fastq.gz
    Renamed 1C5_R1_001.fastq.gz to POC-52-TP4_R1_001.fastq.gz
    Renamed 1C5_R2_001.fastq.gz to POC-52-TP4_R2_001.fastq.gz
    Renamed 1C7_R1_001.fastq.gz to POC-53-TP1_R1_001.fastq.gz
    Renamed 1C7_R2_001.fastq.gz to POC-53-TP1_R2_001.fastq.gz
    Renamed 1D4_R1_001.fastq.gz to POC-52-TP3_R1_001.fastq.gz
    Renamed 1D4_R2_001.fastq.gz to POC-52-TP3_R2_001.fastq.gz
    Renamed 1D5_R1_001.fastq.gz to POC-57-TP1_R1_001.fastq.gz
    Renamed 1D5_R2_001.fastq.gz to POC-57-TP1_R2_001.fastq.gz
    Renamed 1F11_R1_001.fastq.gz to POC-42-TP2_R1_001.fastq.gz
    Renamed 1F11_R2_001.fastq.gz to POC-42-TP2_R2_001.fastq.gz
    Renamed 1F12_R1_001.fastq.gz to POC-255-TP2_R1_001.fastq.gz
    Renamed 1F12_R2_001.fastq.gz to POC-255-TP2_R2_001.fastq.gz
    Renamed 1F2_R1_001.fastq.gz to POC-57-TP3_R1_001.fastq.gz
    Renamed 1F2_R2_001.fastq.gz to POC-57-TP3_R2_001.fastq.gz
    Renamed 1F4_R1_001.fastq.gz to POC-255-TP1_R1_001.fastq.gz
    Renamed 1F4_R2_001.fastq.gz to POC-255-TP1_R2_001.fastq.gz
    Renamed 1F8_R1_001.fastq.gz to POC-53-TP4_R1_001.fastq.gz
    Renamed 1F8_R2_001.fastq.gz to POC-53-TP4_R2_001.fastq.gz
    Renamed 1G12_R1_001.fastq.gz to POC-40-TP1_R1_001.fastq.gz
    Renamed 1G12_R2_001.fastq.gz to POC-40-TP1_R2_001.fastq.gz
    Renamed 1G1_R1_001.fastq.gz to POC-57-TP2_R1_001.fastq.gz
    Renamed 1G1_R2_001.fastq.gz to POC-57-TP2_R2_001.fastq.gz
    Renamed 1G3_R1_001.fastq.gz to POC-259-TP1_R1_001.fastq.gz
    Renamed 1G3_R2_001.fastq.gz to POC-259-TP1_R2_001.fastq.gz
    Renamed 1G7_R1_001.fastq.gz to POC-222-TP4_R1_001.fastq.gz
    Renamed 1G7_R2_001.fastq.gz to POC-222-TP4_R2_001.fastq.gz
    Renamed 1G8_R1_001.fastq.gz to POC-222-TP1_R1_001.fastq.gz
    Renamed 1G8_R2_001.fastq.gz to POC-222-TP1_R2_001.fastq.gz
    Renamed 1H10_R1_001.fastq.gz to POC-219-TP2_R1_001.fastq.gz
    Renamed 1H10_R2_001.fastq.gz to POC-219-TP2_R2_001.fastq.gz
    Renamed 1H1_R1_001.fastq.gz to POC-57-TP4_R1_001.fastq.gz
    Renamed 1H1_R2_001.fastq.gz to POC-57-TP4_R2_001.fastq.gz
    Renamed 1H4_R1_001.fastq.gz to POC-201-TP1_R1_001.fastq.gz
    Renamed 1H4_R2_001.fastq.gz to POC-201-TP1_R2_001.fastq.gz
    Renamed 1H7_R1_001.fastq.gz to POC-219-TP1_R1_001.fastq.gz
    Renamed 1H7_R2_001.fastq.gz to POC-219-TP1_R2_001.fastq.gz
    Renamed 2A1_R1_001.fastq.gz to POC-40-TP4_R1_001.fastq.gz
    Renamed 2A1_R2_001.fastq.gz to POC-40-TP4_R2_001.fastq.gz
    Renamed 2B2_R1_001.fastq.gz to POC-40-TP2_R1_001.fastq.gz
    Renamed 2B2_R2_001.fastq.gz to POC-40-TP2_R2_001.fastq.gz
    Renamed 2D3_R1_001.fastq.gz to POC-52-TP1_R1_001.fastq.gz
    Renamed 2D3_R2_001.fastq.gz to POC-52-TP1_R2_001.fastq.gz
    Renamed 2F1_R1_001.fastq.gz to POC-255-TP3_R1_001.fastq.gz
    Renamed 2F1_R2_001.fastq.gz to POC-255-TP3_R2_001.fastq.gz
    Renamed 2H1_R1_001.fastq.gz to POC-42-TP1_R1_001.fastq.gz
    Renamed 2H1_R2_001.fastq.gz to POC-42-TP1_R2_001.fastq.gz
    Renamed 49--1E12_R1_001.fastq.gz to POC-222-TP2_R1_001.fastq.gz
    Renamed 49--1E12_R2_001.fastq.gz to POC-222-TP2_R2_001.fastq.gz
    Renamed 55--2E3_R1_001.fastq.gz to POC-255-TP4_R1_001.fastq.gz
    Renamed 55--2E3_R2_001.fastq.gz to POC-255-TP4_R2_001.fastq.gz
    Renamed 58--1E10_R1_001.fastq.gz to POC-259-TP3_R1_001.fastq.gz
    Renamed 58--1E10_R2_001.fastq.gz to POC-259-TP3_R2_001.fastq.gz
    Renamed 59--1E1_R1_001.fastq.gz to POC-259-TP4_R1_001.fastq.gz
    Renamed 59--1E1_R2_001.fastq.gz to POC-259-TP4_R2_001.fastq.gz
    Renamed 66--1E2_R1_001.fastq.gz to POC-42-TP3_R1_001.fastq.gz
    Renamed 66--1E2_R2_001.fastq.gz to POC-42-TP3_R2_001.fastq.gz

## 3.1 Count renamed files

``` bash
# Load bash variables into memory
source .bashvars

cd "${raw_reads_dir}"

# Count expected number of P.tuahiniensis samples (assuming each has an R1 file)
expected_count=$(awk -F"," '$9 == "Pocillopora tuahiniensis" { count++ } END {print count}' "${metadata_file}")

# Count the number of R1 fastq files in the raw reads directory
actual_count=$(find "${raw_reads_dir}" -maxdepth 1 -type f -name "POC*_R1_*.fastq.gz" | wc -l)

echo ""
echo "${line}"
echo ""
echo "Expected number of R1 files: ${expected_count}"
echo "Actual number of R1 files: ${actual_count}"
echo ""
echo "${line}"
echo ""

# Compare the counts
if [ "$expected_count" -eq "$actual_count" ]; then
  echo "PASS: All expected R1 files present."
else
  echo "FAIL: Mismatch in expected and actual R1 file counts."
fi
echo ""
echo "${line}"
echo ""
```

    --------------------------------------------------------

    Expected number of R1 files: 39
    Actual number of R1 files: 39

    --------------------------------------------------------

    PASS: All expected R1 files present.

    --------------------------------------------------------

## 3.2 Create MD5 files for new filenames

Although not needed, since the checksums will remain the same, it will
help with legibility by having MD5 checkum files with corresponding
names.

``` bash
# Load bash variables into memory
source .bashvars

cd "${raw_reads_dir}"

for fastq in POC*.fastq.gz
do
  md5sum "${fastq}" | tee --append ${fastq}.md5
done
```

    0e756dbbed6caac6173eaa713755313e  POC-201-TP1_R1_001.fastq.gz
    670a1cdd09afe747f08cb45c7cf691d1  POC-201-TP1_R2_001.fastq.gz
    633f702f89c9fbfc78b417c278444ab4  POC-201-TP2_R1_001.fastq.gz
    86bdc63ad326a45776d6e23d0077572d  POC-201-TP2_R2_001.fastq.gz
    f9a6d80c9e415e2f3573a11ccef2c973  POC-201-TP3_R1_001.fastq.gz
    2ade11ea5883ad71b7af92801b18f6f8  POC-201-TP3_R2_001.fastq.gz
    be8d26b2d04df79cb330beb8bbe686ff  POC-219-TP1_R1_001.fastq.gz
    93ff74adb54112d5643ec502324c6ddc  POC-219-TP1_R2_001.fastq.gz
    bf495e5e64f5ea56880f03bb2b61b2e3  POC-219-TP2_R1_001.fastq.gz
    a324e8de011e15201e03c09d94ff0a68  POC-219-TP2_R2_001.fastq.gz
    9ca1f9199fe74addd8069f0604542e7a  POC-219-TP3_R1_001.fastq.gz
    e55a557dcc9b402dfe330e100f5ae3cf  POC-219-TP3_R2_001.fastq.gz
    2e87ffb757da9929651ab84880e3b34f  POC-219-TP4_R1_001.fastq.gz
    3802cd6c76e48a6f75cd7182491b3750  POC-219-TP4_R2_001.fastq.gz
    c142a93a166883e98b1ba73567209d4b  POC-222-TP1_R1_001.fastq.gz
    da89b618c9a65c767d5a497538cbd19d  POC-222-TP1_R2_001.fastq.gz
    17d87035677f075c6e5afca256156d48  POC-222-TP2_R1_001.fastq.gz
    87835693c200f08b8e13f18cacc1d278  POC-222-TP2_R2_001.fastq.gz
    eafc928eca5f3724ee7892c949317ff3  POC-222-TP3_R1_001.fastq.gz
    065d04621804426bcafd8696d186c511  POC-222-TP3_R2_001.fastq.gz
    d3d8cd4187fcae5e5107218e450e765c  POC-222-TP4_R1_001.fastq.gz
    19f41eabc8d5dcd82465afaa1731876c  POC-222-TP4_R2_001.fastq.gz
    372bb1c02599a9e508d17109ce7480d8  POC-255-TP1_R1_001.fastq.gz
    233ca8d85c8aa9b8b36f163ffd6c7e31  POC-255-TP1_R2_001.fastq.gz
    f70141b32d770e153ee480ab47ff5119  POC-255-TP2_R1_001.fastq.gz
    78025493d09c5d07b03fc73b913c1f2b  POC-255-TP2_R2_001.fastq.gz
    d17a387c326e02c45fdfd2c97846656b  POC-255-TP3_R1_001.fastq.gz
    b50dabc83daf4c7887cdd0327a48847a  POC-255-TP3_R2_001.fastq.gz
    1d73d10f7f2c05daad648468bf2bc64d  POC-255-TP4_R1_001.fastq.gz
    b932b334c5594327e7262ac513d1fd87  POC-255-TP4_R2_001.fastq.gz
    b35b8c2c650e4a9a6038dd1057c5347d  POC-259-TP1_R1_001.fastq.gz
    adcd1b84b878f1508984419b54dc1667  POC-259-TP1_R2_001.fastq.gz
    a3ad83b05a3f7747518353c0b0d4c31d  POC-259-TP2_R1_001.fastq.gz
    9e9c5c42e5c1011251c691aaed9f4411  POC-259-TP2_R2_001.fastq.gz
    8fd0af584a2002b6b8ad85b46c047687  POC-259-TP3_R1_001.fastq.gz
    3ada2039dc28e6e1fe3698e3501b4224  POC-259-TP3_R2_001.fastq.gz
    12464473c7368313bfd86d3a85caac5c  POC-259-TP4_R1_001.fastq.gz
    5c0194c369226bc8d0b9f455df119580  POC-259-TP4_R2_001.fastq.gz
    80f20f5891b9b3a488ea2ddc2d50fe7b  POC-40-TP1_R1_001.fastq.gz
    9372ed756d5afa239a723cfc9d484768  POC-40-TP1_R2_001.fastq.gz
    46b2cd8bcd4b8a66a12b73c9e49fa208  POC-40-TP2_R1_001.fastq.gz
    56e171b37e783dc38ba644a7aab4c105  POC-40-TP2_R2_001.fastq.gz
    79c811fa32d543a3456c3a593f82aca8  POC-40-TP3_R1_001.fastq.gz
    0e55132bf9cd733b0b88cefa3732974b  POC-40-TP3_R2_001.fastq.gz
    077992472930b738f7e2dfa3033191e9  POC-40-TP4_R1_001.fastq.gz
    3c2ac78308b0818e3d689690e02604f0  POC-40-TP4_R2_001.fastq.gz
    e1bfff5c8898dd955547009583cb4248  POC-42-TP1_R1_001.fastq.gz
    90e330a720e5c57fafd5a63b3f718f9e  POC-42-TP1_R2_001.fastq.gz
    f188711d248bf4cf44b438976353904d  POC-42-TP2_R1_001.fastq.gz
    9866519a52134af9a0f3d39e0085b9df  POC-42-TP2_R2_001.fastq.gz
    a19a9f52c2f92c684bf80e677c13f801  POC-42-TP3_R1_001.fastq.gz
    37688ab2a847f3d9961cb9a806ee51ea  POC-42-TP3_R2_001.fastq.gz
    d41c5dbb394baee97000e06821857ce1  POC-42-TP4_R1_001.fastq.gz
    377cc43a38df3a63c9af02cb7a69c42f  POC-42-TP4_R2_001.fastq.gz
    3d1efd40b864bf36226e9ae094b17213  POC-52-TP1_R1_001.fastq.gz
    13f19557b318851400be5b22ca94cf8b  POC-52-TP1_R2_001.fastq.gz
    86eb7e2105b44ca63da2b313c9444a05  POC-52-TP2_R1_001.fastq.gz
    58a0f14831a16f22a2d37de185108102  POC-52-TP2_R2_001.fastq.gz
    cb277d3054080ccb058908157f7979da  POC-52-TP3_R1_001.fastq.gz
    170cc36f7153d9274355d33ddf9b28ef  POC-52-TP3_R2_001.fastq.gz
    60014d094819b98f349903064fbfcb04  POC-52-TP4_R1_001.fastq.gz
    11f41b6708ffd67fd6ae8d971d808d8b  POC-52-TP4_R2_001.fastq.gz
    3504c5e37172a646c43d632716ef660d  POC-53-TP1_R1_001.fastq.gz
    ffb2f33d6285bc373b40451494ca0df3  POC-53-TP1_R2_001.fastq.gz
    176c1c25aadf6dd3ba7a7ea10313ff37  POC-53-TP2_R1_001.fastq.gz
    ec3f3d8f582647c6f8bc8404f2d4b690  POC-53-TP2_R2_001.fastq.gz
    b67053d39541b79c2c408fe712aacc80  POC-53-TP3_R1_001.fastq.gz
    82c42a70aba69cbda906d05b9473b1ac  POC-53-TP3_R2_001.fastq.gz
    8b37441e38687235d9090c413bf5e9ea  POC-53-TP4_R1_001.fastq.gz
    b2509de1e9496d62862e265e98b31c87  POC-53-TP4_R2_001.fastq.gz
    bbe8e779f8d65c4d888418da064d3dce  POC-57-TP1_R1_001.fastq.gz
    7fadf5d1681ab80e4b5294078a22fec1  POC-57-TP1_R2_001.fastq.gz
    69f998f2d05e34df7a2d3aa8615ee2be  POC-57-TP2_R1_001.fastq.gz
    e2adfe081e74540cabbc688216af27f4  POC-57-TP2_R2_001.fastq.gz
    29da3d6b7468166d2574ecfd95962d38  POC-57-TP3_R1_001.fastq.gz
    1e6139e1352b315dde8c0d95078826f3  POC-57-TP3_R2_001.fastq.gz
    be412f3d3ece229e5e9c43492819d476  POC-57-TP4_R1_001.fastq.gz
    7ef22224c5db46506e57cd3e58d86bd8  POC-57-TP4_R2_001.fastq.gz

# 4 FastQC/MultiQC on raw reads

``` bash
# Load bash variables into memory
source .bashvars

############ RUN FASTQC ############


# Create array of trimmed FastQs
raw_fastqs_array=(${raw_reads_dir}/${fastq_pattern})

# Pass array contents to new variable as space-delimited list
raw_fastqc_list=$(echo "${raw_fastqs_array[*]}")

echo "Beginning FastQC on raw reads..."
echo ""

# Run FastQC
### NOTE: Do NOT quote raw_fastqc_list
${programs_array[fastqc]} \
--threads ${threads} \
--outdir ${raw_reads_dir} \
--quiet \
${raw_fastqc_list}

echo "FastQC on raw reads complete!"
echo ""

############ END FASTQC ############

############ RUN MULTIQC ############
echo "Beginning MultiQC on raw FastQC..."
echo ""

${programs_array[multiqc]} \
${raw_reads_dir} \
--interactive \
-o ${raw_reads_dir}

echo ""
echo "MultiQC on raw FastQs complete."
echo ""

############ END MULTIQC ############

echo "Removing FastQC zip files."
echo ""
rm ${raw_reads_dir}/*.zip
echo "FastQC zip files removed."
echo ""

# View directory contents
ls -lh ${raw_reads_dir}
```

    Beginning FastQC on raw reads...

    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    application/gzip
    FastQC on raw reads complete!

    Beginning MultiQC on raw FastQC...


      /// MultiQC 🔍 | v1.14

    |           multiqc | MultiQC Version v1.29 now available!
    |           multiqc | Search path : /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/F-Ptua/data/wgbs-raw-fastqs
    |         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 390/390  
    |            fastqc | Found 78 reports
    |           multiqc | Compressing plot data
    |           multiqc | Report      : ../data/wgbs-raw-fastqs/multiqc_report.html
    |           multiqc | Data        : ../data/wgbs-raw-fastqs/multiqc_data
    |           multiqc | MultiQC complete

    MultiQC on raw FastQs complete.

    Removing FastQC zip files.

    FastQC zip files removed.

    total 202G
    -rw-r--r-- 1 sam sam   57 Dec 30 08:27 1A11_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 08:30 1A11_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 08:44 1A2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 08:48 1A2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 08:57 1A4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 08:59 1A4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 09:42 1B12_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 09:45 1B12_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 10:21 1B7_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 10:23 1B7_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 10:34 1C10_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 10:37 1C10_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 10:40 1C11_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 10:44 1C11_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 10:46 1C12_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 10:48 1C12_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 10:51 1C1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 10:54 1C1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 11:01 1C3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 11:04 1C3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 11:06 1C4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 11:08 1C4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 11:10 1C5_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 11:13 1C5_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 11:20 1C7_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 11:23 1C7_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 12:07 1D4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 12:10 1D4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 12:12 1D5_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 12:14 1D5_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 12:41 1F11_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 12:44 1F11_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 12:46 1F12_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 12:48 1F12_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 12:56 1F2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 12:58 1F2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 13:04 1F4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 13:07 1F4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 13:23 1F8_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 13:26 1F8_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 13:41 1G12_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 13:44 1G12_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 13:46 1G1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 13:49 1G1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 13:57 1G3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 13:59 1G3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 14:14 1G7_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 14:17 1G7_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 14:20 1G8_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 14:22 1G8_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 14:30 1H10_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   57 Dec 30 14:33 1H10_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 14:46 1H1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 14:48 1H1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 15:01 1H4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 15:04 1H4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 15:19 1H7_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 15:22 1H7_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 15:36 2A1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 15:38 2A1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 15:54 2B2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 15:57 2B2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 16:32 2D3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 16:34 2D3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 16:37 2F1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 16:40 2F1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 17:10 2H1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   56 Dec 30 17:14 2H1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   61 Dec 30 17:32 49--1E12_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   61 Dec 30 17:35 49--1E12_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   60 Dec 30 17:38 55--2E3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   60 Dec 30 17:41 55--2E3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   61 Dec 30 17:44 58--1E10_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   61 Dec 30 17:47 58--1E10_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   60 Dec 30 17:50 59--1E1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   60 Dec 30 17:54 59--1E1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   60 Dec 30 18:03 66--1E2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam   60 Dec 30 18:06 66--1E2_R2_001.fastq.gz.md5
    drwxr-xr-x 2 sam sam 4.0K May 28 10:24 multiqc_data
    -rw-r--r-- 1 sam sam 2.1M May 28 10:24 multiqc_report.html
    -rw-r--r-- 1 sam sam 586K May 28 10:14 POC-201-TP1_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 15:01 POC-201-TP1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:45 POC-201-TP1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 605K May 28 10:14 POC-201-TP1_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 15:04 POC-201-TP1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:45 POC-201-TP1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 588K May 28 10:11 POC-201-TP2_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.0G Dec 30 11:06 POC-201-TP2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:45 POC-201-TP2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 606K May 28 10:11 POC-201-TP2_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.0G Dec 30 11:08 POC-201-TP2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:45 POC-201-TP2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 590K May 28 10:13 POC-201-TP3_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.8G Dec 30 08:27 POC-201-TP3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:46 POC-201-TP3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 613K May 28 10:14 POC-201-TP3_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.8G Dec 30 08:30 POC-201-TP3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:46 POC-201-TP3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 587K May 28 10:12 POC-219-TP1_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.3G Dec 30 15:19 POC-219-TP1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:46 POC-219-TP1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 608K May 28 10:12 POC-219-TP1_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.2G Dec 30 15:22 POC-219-TP1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:46 POC-219-TP1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 588K May 28 10:17 POC-219-TP2_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 3.3G Dec 30 14:30 POC-219-TP2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:46 POC-219-TP2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 609K May 28 10:17 POC-219-TP2_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 3.3G Dec 30 14:33 POC-219-TP2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:46 POC-219-TP2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 594K May 28 10:16 POC-219-TP3_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 3.0G Dec 30 08:44 POC-219-TP3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:47 POC-219-TP3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 617K May 28 10:15 POC-219-TP3_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 3.1G Dec 30 08:48 POC-219-TP3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:47 POC-219-TP3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 587K May 28 10:13 POC-219-TP4_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.7G Dec 30 09:42 POC-219-TP4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:47 POC-219-TP4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 609K May 28 10:14 POC-219-TP4_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.7G Dec 30 09:45 POC-219-TP4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:48 POC-219-TP4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 583K May 28 10:14 POC-222-TP1_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.7G Dec 30 14:20 POC-222-TP1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:48 POC-222-TP1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 608K May 28 10:14 POC-222-TP1_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.7G Dec 30 14:22 POC-222-TP1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:48 POC-222-TP1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 587K May 28 10:12 POC-222-TP2_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.2G Dec 30 17:32 POC-222-TP2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:48 POC-222-TP2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 604K May 28 10:12 POC-222-TP2_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.2G Dec 30 17:35 POC-222-TP2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:48 POC-222-TP2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 588K May 28 10:16 POC-222-TP3_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 3.3G Dec 30 10:51 POC-222-TP3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:48 POC-222-TP3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 606K May 28 10:16 POC-222-TP3_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 3.3G Dec 30 10:54 POC-222-TP3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:48 POC-222-TP3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 589K May 28 10:13 POC-222-TP4_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.9G Dec 30 14:14 POC-222-TP4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:48 POC-222-TP4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 610K May 28 10:12 POC-222-TP4_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.8G Dec 30 14:17 POC-222-TP4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:48 POC-222-TP4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 585K May 28 10:11 POC-255-TP1_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.5G Dec 30 13:04 POC-255-TP1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:49 POC-255-TP1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 608K May 28 10:11 POC-255-TP1_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 13:07 POC-255-TP1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:49 POC-255-TP1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 590K May 28 10:11 POC-255-TP2_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.4G Dec 30 12:46 POC-255-TP2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:49 POC-255-TP2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 610K May 28 10:11 POC-255-TP2_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.4G Dec 30 12:48 POC-255-TP2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:49 POC-255-TP2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 598K May 28 10:12 POC-255-TP3_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 16:37 POC-255-TP3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:49 POC-255-TP3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 603K May 28 10:12 POC-255-TP3_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.7G Dec 30 16:40 POC-255-TP3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:49 POC-255-TP3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 588K May 28 10:12 POC-255-TP4_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.5G Dec 30 17:38 POC-255-TP4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:49 POC-255-TP4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 597K May 28 10:12 POC-255-TP4_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.5G Dec 30 17:41 POC-255-TP4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:49 POC-255-TP4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 586K May 28 10:12 POC-259-TP1_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.7G Dec 30 13:57 POC-259-TP1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:49 POC-259-TP1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 602K May 28 10:12 POC-259-TP1_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.7G Dec 30 13:59 POC-259-TP1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:49 POC-259-TP1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 583K May 28 10:15 POC-259-TP2_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 3.0G Dec 30 08:57 POC-259-TP2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:50 POC-259-TP2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 605K May 28 10:15 POC-259-TP2_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.9G Dec 30 08:59 POC-259-TP2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:50 POC-259-TP2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 590K May 28 10:13 POC-259-TP3_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 3.0G Dec 30 17:44 POC-259-TP3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:50 POC-259-TP3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 610K May 28 10:14 POC-259-TP3_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 3.0G Dec 30 17:47 POC-259-TP3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:50 POC-259-TP3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 585K May 28 10:13 POC-259-TP4_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.9G Dec 30 17:50 POC-259-TP4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:50 POC-259-TP4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 610K May 28 10:13 POC-259-TP4_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.9G Dec 30 17:54 POC-259-TP4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   62 May 28 09:50 POC-259-TP4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 586K May 28 10:10 POC-40-TP1_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.2G Dec 30 13:41 POC-40-TP1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:50 POC-40-TP1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 606K May 28 10:11 POC-40-TP1_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.2G Dec 30 13:44 POC-40-TP1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:51 POC-40-TP1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 580K May 28 10:22 POC-40-TP2_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.9G Dec 30 15:54 POC-40-TP2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:51 POC-40-TP2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 585K May 28 10:22 POC-40-TP2_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.9G Dec 30 15:57 POC-40-TP2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:51 POC-40-TP2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 584K May 28 10:22 POC-40-TP3_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 10:21 POC-40-TP3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:51 POC-40-TP3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 606K May 28 10:21 POC-40-TP3_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 10:23 POC-40-TP3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:51 POC-40-TP3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 577K May 28 10:21 POC-40-TP4_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.4G Dec 30 15:36 POC-40-TP4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:51 POC-40-TP4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 589K May 28 10:22 POC-40-TP4_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.4G Dec 30 15:38 POC-40-TP4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:52 POC-40-TP4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 601K May 28 10:23 POC-42-TP1_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.7G Dec 30 17:10 POC-42-TP1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:52 POC-42-TP1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 604K May 28 10:23 POC-42-TP1_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.8G Dec 30 17:14 POC-42-TP1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:52 POC-42-TP1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 587K May 28 10:23 POC-42-TP2_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.9G Dec 30 12:41 POC-42-TP2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:52 POC-42-TP2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 606K May 28 10:23 POC-42-TP2_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.9G Dec 30 12:44 POC-42-TP2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:52 POC-42-TP2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 587K May 28 10:22 POC-42-TP3_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 18:03 POC-42-TP3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:52 POC-42-TP3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 609K May 28 10:22 POC-42-TP3_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 18:06 POC-42-TP3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:52 POC-42-TP3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 583K May 28 10:22 POC-42-TP4_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.8G Dec 30 10:40 POC-42-TP4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:52 POC-42-TP4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 605K May 28 10:22 POC-42-TP4_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.8G Dec 30 10:44 POC-42-TP4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:53 POC-42-TP4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 582K May 28 10:20 POC-52-TP1_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 1.8G Dec 30 16:32 POC-52-TP1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:53 POC-52-TP1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 593K May 28 10:20 POC-52-TP1_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 1.8G Dec 30 16:34 POC-52-TP1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:53 POC-52-TP1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 580K May 28 10:22 POC-52-TP2_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.3G Dec 30 10:46 POC-52-TP2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:53 POC-52-TP2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 607K May 28 10:21 POC-52-TP2_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.2G Dec 30 10:48 POC-52-TP2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:53 POC-52-TP2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 585K May 28 10:23 POC-52-TP3_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 3.1G Dec 30 12:07 POC-52-TP3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:54 POC-52-TP3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 607K May 28 10:24 POC-52-TP3_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 3.1G Dec 30 12:10 POC-52-TP3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:54 POC-52-TP3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 589K May 28 10:22 POC-52-TP4_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.2G Dec 30 11:10 POC-52-TP4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:54 POC-52-TP4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 611K May 28 10:22 POC-52-TP4_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.2G Dec 30 11:13 POC-52-TP4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:54 POC-52-TP4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 586K May 28 10:23 POC-53-TP1_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.8G Dec 30 11:20 POC-53-TP1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:55 POC-53-TP1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 610K May 28 10:23 POC-53-TP1_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.8G Dec 30 11:23 POC-53-TP1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:55 POC-53-TP1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 587K May 28 10:23 POC-53-TP2_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.8G Dec 30 10:34 POC-53-TP2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:55 POC-53-TP2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 606K May 28 10:23 POC-53-TP2_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.8G Dec 30 10:37 POC-53-TP2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:56 POC-53-TP2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 586K May 28 10:23 POC-53-TP3_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 11:01 POC-53-TP3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:56 POC-53-TP3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 607K May 28 10:23 POC-53-TP3_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 11:04 POC-53-TP3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:56 POC-53-TP3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 588K May 28 10:23 POC-53-TP4_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.5G Dec 30 13:23 POC-53-TP4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:57 POC-53-TP4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 610K May 28 10:23 POC-53-TP4_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.5G Dec 30 13:26 POC-53-TP4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:57 POC-53-TP4_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 587K May 28 10:23 POC-57-TP1_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.2G Dec 30 12:12 POC-57-TP1_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:57 POC-57-TP1_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 611K May 28 10:22 POC-57-TP1_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.2G Dec 30 12:14 POC-57-TP1_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:57 POC-57-TP1_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 587K May 28 10:24 POC-57-TP2_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.8G Dec 30 13:46 POC-57-TP2_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:58 POC-57-TP2_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 607K May 28 10:23 POC-57-TP2_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.7G Dec 30 13:49 POC-57-TP2_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:58 POC-57-TP2_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 588K May 28 10:23 POC-57-TP3_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 12:56 POC-57-TP3_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:58 POC-57-TP3_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 609K May 28 10:23 POC-57-TP3_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.6G Dec 30 12:58 POC-57-TP3_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:59 POC-57-TP3_R2_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 588K May 28 10:23 POC-57-TP4_R1_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.5G Dec 30 14:46 POC-57-TP4_R1_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:59 POC-57-TP4_R1_001.fastq.gz.md5
    -rw-r--r-- 1 sam sam 612K May 28 10:23 POC-57-TP4_R2_001_fastqc.html
    -rw-r--r-- 1 sam sam 2.5G Dec 30 14:48 POC-57-TP4_R2_001.fastq.gz
    -rw-r--r-- 1 sam sam   61 May 28 09:59 POC-57-TP4_R2_001.fastq.gz.md5

<div id="refs" class="references csl-bib-body hanging-indent">

<div id="ref-ewels2016" class="csl-entry">

Ewels, Philip, Måns Magnusson, Sverker Lundin, and Max Käller. 2016.
“MultiQC: Summarize Analysis Results for Multiple Tools and Samples in a
Single Report.” *Bioinformatics* 32 (19): 3047–48.
<https://doi.org/10.1093/bioinformatics/btw354>.

</div>

</div>
