### Background 

This code will align trimmed _P.tuahiniensis_ RNA-seq data to the _P.tuahiniensis_ genome using STAR and stringtie to create a gene and transcript counts matrices using the [nf-core RNAseq pipeline](https://nf-co.re/rnaseq/3.18.0/.) 


**Input data:**

- [Trimmed fastqs](https://gannet.fish.washington.edu/gitrepos/urol-e5/timeseries_molecular/F-Ptua/output/01.00-F-Ptua-RNAseq-trimming-fastp-FastQC-MultiQC/)
- [genome](https://owl.fish.washington.edu/halfshell/genomic-databank/Pocillopora_meandrina_HIv1.assembly.fasta)
- [gtf](https://github.com/urol-e5/timeseries_molecular/raw/d5f546705e3df40558eeaa5c18b122c79d2f4453/F-Ptua/data/Pocillopora_meandrina_HIv1.genes-validated.gtf)
- [gff](https://github.com/urol-e5/timeseries_molecular/raw/d5f546705e3df40558eeaa5c18b122c79d2f4453/F-Ptua/data/Pocillopora_meandrina_HIv1.genes-validated.gff3)

**Files needed to run nextflow pipeline:**

- [config file](https://gannet.fish.washington.edu/metacarcinus/E5/Ptuahiniensis/20250421_RNAseq/uw_hyak_srlab.config)
- [samplesheet](https://gannet.fish.washington.edu/metacarcinus/E5/Ptuahiniensis/20250421_RNAseq/samplesheet.csv)

### Methods
**Copy genome files to klone**

```
# show path
pwd
/gscratch/srlab/strigg/GENOMES

# copy genome
wget https://owl.fish.washington.edu/halfshell/genomic-databank/Pocillopora_meandrina_HIv1.assembly.fasta

#copy gtf file
wget https://github.com/urol-e5/timeseries_molecular/raw/d5f546705e3df40558eeaa5c18b122c79d2f4453/F-Ptua/data/Pocillopora_meandrina_HIv1.genes-validated.gtf

# copy gff file
wget https://github.com/urol-e5/timeseries_molecular/raw/d5f546705e3df40558eeaa5c18b122c79d2f4453/F-Ptua/data/Pocillopora_meandrina_HIv1.genes-validated.gff3
```


**Copy RNAseq data to Klone**

```
# open screen session (reopened existing session)
screen -S RNAseq

# start interactive node
salloc -A srlab -p cpu-g2-mem2x -N 1 -c 1 --mem=16GB --time=16:00:00

# copy data
rsync --progress --verbose --archive shellytrigg@gannet.fish.washington.edu:/volume2/web/gitrepos/urol-e5/timeseries_molecular/F-Ptua/output/01.00-F-Ptua-RNAseq-trimming-fastp-FastQC-MultiQC/*.gz /gscratch/scrubbed/strigg/analyses/20250421_RNAseq
```

**Run nf-core RNA-seq pipeline on Klone**

```
# open screen session 
screen -S RNAseq

# start interactive node
salloc -A srlab -p cpu-g2-mem2x -N 1 -c 1 --mem=16GB --time=24:00:00

# activate conda environment. Note: you can use `conda` instead of `mamba` 
mamba activate /gscratch/srlab/nextflow/bin/miniforge/envs/nextflow

# run pipeline
nextflow run nf-core/rnaseq -resume \
-c /gscratch/srlab/nextflow/uw_hyak_srlab.config \
--input /gscratch/scrubbed/strigg/analyses/20250421_RNAseq/samplesheet.csv \
--outdir /gscratch/scrubbed/strigg/analyses/20250421_RNAseq \
--gtf /gscratch/srlab/strigg/GENOMES/Pocillopora_meandrina_HIv1.genes-validated.gtf \
--gff /gscratch/srlab/strigg/GENOMES/Pocillopora_meandrina_HIv1.genes-validated.gff.gz \
--fasta /gscratch/srlab/strigg/GENOMES/Pocillopora_meandrina_HIv1.assembly.fasta \
--skip_trimming \
--aligner star_salmon \
--skip_pseudo_alignment \
--multiqc_title Pmeandrina_RNAseq \
--deseq2_vst

```

### Results

- Multiqc report: [https://gannet.fish.washington.edu/metacarcinus/E5/Ptuahiniensis/20250421_RNAseq/multiqc/multiqc_report.html](https://gannet.fish.washington.edu/metacarcinus/E5/Ptuahiniensis/20250421_RNAseq/multiqc/multiqc_report.html)
- Pipeline report: [https://gannet.fish.washington.edu/metacarcinus/E5/Ptuahiniensis/20250421_RNAseq/pipeline_info/execution_report_2025-04-21_10-32-32.html](https://gannet.fish.washington.edu/metacarcinus/E5/Ptuahiniensis/20250421_RNAseq/pipeline_info/execution_report_2025-04-21_10-32-32.html)
- Pipeline timeline: [https://gannet.fish.washington.edu/metacarcinus/E5/Ptuahiniensis/20250421_RNAseq/pipeline_info/execution_timeline_2025-04-21_10-32-32.html](https://gannet.fish.washington.edu/metacarcinus/E5/Ptuahiniensis/20250421_RNAseq/pipeline_info/execution_timeline_2025-04-21_10-32-32.html)
- Counts matrices: [https://gannet.fish.washington.edu/metacarcinus/E5/Ptuahiniensis/20250421_RNAseq/](https://gannet.fish.washington.edu/metacarcinus/E5/Ptuahiniensis/20250421_RNAseq/)
	- salmon.merged.gene_counts.tsv
	- salmon.merged.gene\_counts\_legnth_scaled.tsv
	- salmon.merged.gene_counts\_scaled.tsv
	- salmon.merged.gene_lengths.tsv
	- salmon.merged.gene_tpm.tsv
	- salmon.merged.transcript_counts.tsv
	- salmon.merged.transcript_lengths.tsv
	- salmon.merged.transcript_tpm.tsv
